// react-rainbow-components/components/MarkdownOutput/renderers/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireDefault(require("react"));

var _styled = require("./styled");

/* eslint-disable react/prop-types, id-length, react/jsx-props-no-spreading */
var Heading = function Heading(_ref) {
  var level = _ref.level,
      children = _ref.children;
  return /*#__PURE__*/_react["default"].createElement(_styled.StyledHeading, {
    as: "h".concat(level),
    level: level
  }, children);
};

var _default = {
  h1: function h1(props) {
    return /*#__PURE__*/_react["default"].createElement(Heading, (0, _extends2["default"])({
      level: 1
    }, props));
  },
  h2: function h2(props) {
    return /*#__PURE__*/_react["default"].createElement(Heading, (0, _extends2["default"])({
      level: 2
    }, props));
  },
  h3: function h3(props) {
    return /*#__PURE__*/_react["default"].createElement(Heading, (0, _extends2["default"])({
      level: 3
    }, props));
  },
  h4: function h4(props) {
    return /*#__PURE__*/_react["default"].createElement(Heading, (0, _extends2["default"])({
      level: 4
    }, props));
  },
  h5: function h5(props) {
    return /*#__PURE__*/_react["default"].createElement(Heading, (0, _extends2["default"])({
      level: 5
    }, props));
  },
  h6: function h6(props) {
    return /*#__PURE__*/_react["default"].createElement(Heading, (0, _extends2["default"])({
      level: 6
    }, props));
  },
  code: _styled.StyledCode,
  p: _styled.StyledParagraph,
  hr: _styled.StyledHR,
  ul: _styled.StyledList,
  ol: _styled.StyledOrderedList,
  li: _styled.StyledListItem,
  table: _styled.StyledTable,
  tbody: _styled.StyledTableBody,
  th: _styled.StyledTableHeadindCell,
  tr: _styled.StyledTableRow,
  td: _styled.StyledTableCell
};
exports["default"] = _default;
