// react-rainbow-components/components/PhoneInput/countries.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var Flags = _interopRequireWildcard(require("./flags"));

/* eslintDisable global-require */
var style = {
  width: '28px',
  height: '28px'
};
var _default = [{
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Afghanistan, {
    style: style
  }),
  countryCode: '+93',
  country: 'Afghanistan',
  isoCode: 'af'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.AlandIslands, {
    style: style
  }),
  countryCode: '+358',
  country: 'Åland islands',
  isoCode: 'ax'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Albania, {
    style: style
  }),
  countryCode: '+355',
  country: 'Albania',
  isoCode: 'al'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Algeria, {
    style: style
  }),
  countryCode: '+213',
  country: 'Algeria',
  isoCode: 'dz'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.AmericanSamoa, {
    style: style
  }),
  countryCode: '+1',
  country: 'American Somoa',
  isoCode: 'as'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Andorra, {
    style: style
  }),
  countryCode: '+376',
  country: 'Andorra',
  isoCode: 'ad'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Angola, {
    style: style
  }),
  countryCode: '+244',
  country: 'Angola',
  isoCode: 'ao'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Anguilla, {
    style: style
  }),
  countryCode: '+1',
  country: 'Anguilla',
  isoCode: 'ai'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.AntiguaAndBarbuda, {
    style: style
  }),
  countryCode: '+1',
  country: 'Antigua and Barbuda',
  isoCode: 'ag'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Argentina, {
    style: style
  }),
  countryCode: '+54',
  country: 'Argentina',
  isoCode: 'ar'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Armenia, {
    style: style
  }),
  countryCode: '+374',
  country: 'Armenia',
  isoCode: 'am'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Aruba, {
    style: style
  }),
  countryCode: '+297',
  country: 'Aruba',
  isoCode: 'aw'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.AscensionIsland, {
    style: style
  }),
  countryCode: '+247',
  country: 'Ascension Island',
  isoCode: 'ac'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Australia, {
    style: style
  }),
  countryCode: '+61',
  country: 'Australia',
  isoCode: 'au'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Austria, {
    style: style
  }),
  countryCode: '+43',
  country: 'Austria',
  isoCode: 'at'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Azerbaijan, {
    style: style
  }),
  countryCode: '+994',
  country: 'Azerbaijan',
  isoCode: 'az'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Bahamas, {
    style: style
  }),
  countryCode: '+1',
  country: 'Bahamas',
  isoCode: 'bs'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Bahrain, {
    style: style
  }),
  countryCode: '+973',
  country: 'Bahrain',
  isoCode: 'bh'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Bangladesh, {
    style: style
  }),
  countryCode: '+880',
  country: 'Bangladesh',
  isoCode: 'bd'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Barbados, {
    style: style
  }),
  countryCode: '+1',
  country: 'Barbados',
  isoCode: 'bb'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Belarus, {
    style: style
  }),
  countryCode: '+375',
  country: 'Belarus',
  isoCode: 'by'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Belgium, {
    style: style
  }),
  countryCode: '+32',
  country: 'Belgium',
  isoCode: 'be'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Belize, {
    style: style
  }),
  countryCode: '+501',
  country: 'Belize',
  isoCode: 'bz'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Benin, {
    style: style
  }),
  countryCode: '+229',
  country: 'Benin',
  isoCode: 'bj'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Bermuda, {
    style: style
  }),
  countryCode: '+1',
  country: 'Bermuda',
  isoCode: 'bm'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Bhutan, {
    style: style
  }),
  countryCode: '+975',
  country: 'Bhutan',
  isoCode: 'bt'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Bolivia, {
    style: style
  }),
  countryCode: '+591',
  country: 'Bolivia',
  isoCode: 'bo'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Bonaire, {
    style: style
  }),
  countryCode: '+599',
  country: 'Bonaire',
  isoCode: 'bq'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.BosniaAndHerzegovina, {
    style: style
  }),
  countryCode: '+387',
  country: 'Bosnia and Herzegovina',
  isoCode: 'ba'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Botswana, {
    style: style
  }),
  countryCode: '+267',
  country: 'Botswana',
  isoCode: 'bw'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Brazil, {
    style: style
  }),
  countryCode: '+55',
  country: 'Brazil',
  isoCode: 'br'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.BritishVirginIslands, {
    style: style
  }),
  countryCode: '+1',
  country: 'British Virgin Islands',
  isoCode: 'vg'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Brunei, {
    style: style
  }),
  countryCode: '+673',
  country: 'Brunei',
  isoCode: 'bn'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Bulgaria, {
    style: style
  }),
  countryCode: '+359',
  country: 'Bulgaria',
  isoCode: 'bg'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.BurkinaFaso, {
    style: style
  }),
  countryCode: '+226',
  country: 'Burkina Faso',
  isoCode: 'bf'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Burundi, {
    style: style
  }),
  countryCode: '+257',
  country: 'Burundi',
  isoCode: 'bi'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Cambodia, {
    style: style
  }),
  countryCode: '+855',
  country: 'Cambodia',
  isoCode: 'kh'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Cameroon, {
    style: style
  }),
  countryCode: '+237',
  country: 'Cameroon',
  isoCode: 'cm'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Canada, {
    style: style
  }),
  countryCode: '+1',
  country: 'Canada',
  isoCode: 'ca'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.CapeVerde, {
    style: style
  }),
  countryCode: '+238',
  country: 'Cape Verde',
  isoCode: 'cv'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.CaymanIslands, {
    style: style
  }),
  countryCode: '+1',
  country: 'Cayman Islands',
  isoCode: 'ky'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.CentralAfricanRepublic, {
    style: style
  }),
  countryCode: '+236',
  country: 'Central African Republic',
  isoCode: 'cf'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Chad, {
    style: style
  }),
  countryCode: '+235',
  country: 'Chad',
  isoCode: 'td'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Chile, {
    style: style
  }),
  countryCode: '+56',
  country: 'Chile',
  isoCode: 'cl'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.China, {
    style: style
  }),
  countryCode: '+86',
  country: 'China',
  isoCode: 'cn'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.ChristmasIsland, {
    style: style
  }),
  countryCode: '+61',
  country: 'Christmas Island',
  isoCode: 'cx'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.CocosIsland, {
    style: style
  }),
  countryCode: '+61',
  country: 'Cocos Island',
  isoCode: 'cc'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Colombia, {
    style: style
  }),
  countryCode: '+57',
  country: 'Colombia',
  isoCode: 'co'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Comoros, {
    style: style
  }),
  countryCode: '+269',
  country: 'Comoros',
  isoCode: 'km'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.CongoBrazzaville, {
    style: style
  }),
  countryCode: '+242',
  country: 'Congo - Brazzaville',
  isoCode: 'cg'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.CongoKinshasa, {
    style: style
  }),
  countryCode: '+243',
  country: 'Congo - Kinshasa',
  isoCode: 'cd'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.CookIslands, {
    style: style
  }),
  countryCode: '+682',
  country: 'Cook Islands',
  isoCode: 'ck'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.CostaRica, {
    style: style
  }),
  countryCode: '+506',
  country: 'Costa Rica',
  isoCode: 'cr'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.CoteDLvoire, {
    style: style
  }),
  countryCode: '+225',
  country: 'Côte d’lvoire',
  isoCode: 'ci'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Croatia, {
    style: style
  }),
  countryCode: '+385',
  country: 'Croatia',
  isoCode: 'hr'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Cuba, {
    style: style
  }),
  countryCode: '+53',
  country: 'Cuba',
  isoCode: 'cu'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Curacao, {
    style: style
  }),
  countryCode: '+599',
  country: 'Curaçao',
  isoCode: 'cw'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Cyprus, {
    style: style
  }),
  countryCode: '+357',
  country: 'Cyprus',
  isoCode: 'cy'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.CzechRepublic, {
    style: style
  }),
  countryCode: '+420',
  country: 'Czech Republic',
  isoCode: 'cz'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Denmark, {
    style: style
  }),
  countryCode: '+45',
  country: 'Denmark',
  isoCode: 'dk'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Djibouti, {
    style: style
  }),
  countryCode: '+253',
  country: 'Djibouti',
  isoCode: 'dj'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Dominica, {
    style: style
  }),
  countryCode: '+1',
  country: 'Dominica',
  isoCode: 'dm'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.DominicanRepublic, {
    style: style
  }),
  countryCode: '+1',
  country: 'Dominican Republic',
  isoCode: 'do'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Ecuador, {
    style: style
  }),
  countryCode: '+593',
  country: 'Ecuador',
  isoCode: 'ec'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Egypt, {
    style: style
  }),
  countryCode: '+20',
  country: 'Egypt',
  isoCode: 'eg'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.ElSalvador, {
    style: style
  }),
  countryCode: '+503',
  country: 'El Salvador',
  isoCode: 'sv'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.EquatorialGuinea, {
    style: style
  }),
  countryCode: '+240',
  country: 'Equatorial Guinea',
  isoCode: 'gq'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Eritrea, {
    style: style
  }),
  countryCode: '+291',
  country: 'Eritrea',
  isoCode: 'er'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Estonia, {
    style: style
  }),
  countryCode: '+372',
  country: 'Estonia',
  isoCode: 'ee'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Ethiopia, {
    style: style
  }),
  countryCode: '+251',
  country: 'Ethiopia',
  isoCode: 'et'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.FalklandIslands, {
    style: style
  }),
  countryCode: '+500',
  country: 'Falkland Islands',
  isoCode: 'fk'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.FaroeIslands, {
    style: style
  }),
  countryCode: '+298',
  country: 'Faroe Islands',
  isoCode: 'fo'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Fiji, {
    style: style
  }),
  countryCode: '+679',
  country: 'Fiji',
  isoCode: 'fj'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Finland, {
    style: style
  }),
  countryCode: '+358',
  country: 'Finland',
  isoCode: 'fi'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.France, {
    style: style
  }),
  countryCode: '+33',
  country: 'France',
  isoCode: 'fr'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.FrenchGuiana, {
    style: style
  }),
  countryCode: '+594',
  country: 'French Guiana',
  isoCode: 'gf'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.FrenchPolynesia, {
    style: style
  }),
  countryCode: '+689',
  country: 'French Polynesia',
  isoCode: 'pf'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Gabon, {
    style: style
  }),
  countryCode: '+241',
  country: 'Gabon',
  isoCode: 'ga'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Gambia, {
    style: style
  }),
  countryCode: '+220',
  country: 'Gambia',
  isoCode: 'gm'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Georgia, {
    style: style
  }),
  countryCode: '+995',
  country: 'Georgia',
  isoCode: 'ge'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Germany, {
    style: style
  }),
  countryCode: '+49',
  country: 'Germany',
  isoCode: 'de'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Ghana, {
    style: style
  }),
  countryCode: '+233',
  country: 'Ghana',
  isoCode: 'gh'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Gibraltar, {
    style: style
  }),
  countryCode: '+350',
  country: 'Gibraltar',
  isoCode: 'gi'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Greece, {
    style: style
  }),
  countryCode: '+30',
  country: 'Greece',
  isoCode: 'gr'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Greenland, {
    style: style
  }),
  countryCode: '+299',
  country: 'Greenland',
  isoCode: 'gl'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Grenada, {
    style: style
  }),
  countryCode: '+1',
  country: 'Grenada',
  isoCode: 'gd'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Guadeloupe, {
    style: style
  }),
  countryCode: '+590',
  country: 'Guadeloupe',
  isoCode: 'gp'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Guam, {
    style: style
  }),
  countryCode: '+1',
  country: 'Guam',
  isoCode: 'gu'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Guatemala, {
    style: style
  }),
  countryCode: '+502',
  country: 'Guatemala',
  isoCode: 'gt'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Guernsey, {
    style: style
  }),
  countryCode: '+44',
  country: 'Guernsey',
  isoCode: 'gg'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Guinea, {
    style: style
  }),
  countryCode: '+224',
  country: 'Guinea',
  isoCode: 'gn'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.GuineaBissau, {
    style: style
  }),
  countryCode: '+245',
  country: 'Guinea Bissau',
  isoCode: 'gw'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Guyana, {
    style: style
  }),
  countryCode: '+592',
  country: 'Guyana',
  isoCode: 'gy'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Haiti, {
    style: style
  }),
  countryCode: '+509',
  country: 'Haiti',
  isoCode: 'ht'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Honduras, {
    style: style
  }),
  countryCode: '+504',
  country: 'Honduras',
  isoCode: 'hn'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.HongKongSarChina, {
    style: style
  }),
  countryCode: '+852',
  country: 'Hong Kong SAR China',
  isoCode: 'hk'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Hungary, {
    style: style
  }),
  countryCode: '+36',
  country: 'Hungary',
  isoCode: 'hu'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Iceland, {
    style: style
  }),
  countryCode: '+354',
  country: 'Iceland',
  isoCode: 'is'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.India, {
    style: style
  }),
  countryCode: '+91',
  country: 'India',
  isoCode: 'in'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Indonesia, {
    style: style
  }),
  countryCode: '+62',
  country: 'Indonesia',
  isoCode: 'id'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Iran, {
    style: style
  }),
  countryCode: '+98',
  country: 'Iran',
  isoCode: 'ir'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Iraq, {
    style: style
  }),
  countryCode: '+964',
  country: 'Iraq',
  isoCode: 'iq'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Ireland, {
    style: style
  }),
  countryCode: '+353',
  country: 'Ireland',
  isoCode: 'ie'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.IsleOfMan, {
    style: style
  }),
  countryCode: '+44',
  country: 'Isle of Man',
  isoCode: 'im'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Israel, {
    style: style
  }),
  countryCode: '+972',
  country: 'Israel',
  isoCode: 'il'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Italy, {
    style: style
  }),
  countryCode: '+39',
  country: 'Italy',
  isoCode: 'it'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Jamaica, {
    style: style
  }),
  countryCode: '+1',
  country: 'Jamaica',
  isoCode: 'jm'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Japan, {
    style: style
  }),
  countryCode: '+81',
  country: 'Japan',
  isoCode: 'jp'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Jersey, {
    style: style
  }),
  countryCode: '+44',
  country: 'Jersey',
  isoCode: 'je'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Jordan, {
    style: style
  }),
  countryCode: '+962',
  country: 'Jordan',
  isoCode: 'jo'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Kazakhstan, {
    style: style
  }),
  countryCode: '+7',
  country: 'Kazakhstan',
  isoCode: 'kz'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Kenya, {
    style: style
  }),
  countryCode: '+254',
  country: 'Kenya',
  isoCode: 'ke'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Kiribati, {
    style: style
  }),
  countryCode: '+686',
  country: 'Kiribati',
  isoCode: 'ki'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Kosovo, {
    style: style
  }),
  countryCode: '+383',
  country: 'Kosovo',
  isoCode: 'kv'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Kuwait, {
    style: style
  }),
  countryCode: '+965',
  country: 'Kuwait',
  isoCode: 'kw'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Kyrgyzstan, {
    style: style
  }),
  countryCode: '+996',
  country: 'Kyrgyzstan',
  isoCode: 'kg'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Laos, {
    style: style
  }),
  countryCode: '+856',
  country: 'Laos',
  isoCode: 'la'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Latvia, {
    style: style
  }),
  countryCode: '+371',
  country: 'Latvia',
  isoCode: 'lv'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Lebanon, {
    style: style
  }),
  countryCode: '+961',
  country: 'Lebanon',
  isoCode: 'lb'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Lesotho, {
    style: style
  }),
  countryCode: '+266',
  country: 'Lesotho',
  isoCode: 'ls'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Liberia, {
    style: style
  }),
  countryCode: '+231',
  country: 'Liberia',
  isoCode: 'lr'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Libya, {
    style: style
  }),
  countryCode: '+218',
  country: 'Libya',
  isoCode: 'ly'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Liechtenstein, {
    style: style
  }),
  countryCode: '+423',
  country: 'Liechtenstein',
  isoCode: 'li'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Lithuania, {
    style: style
  }),
  countryCode: '+370',
  country: 'Lithuania',
  isoCode: 'lt'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Luxembourg, {
    style: style
  }),
  countryCode: '+352',
  country: 'Luxembourg',
  isoCode: 'lu'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.MacauSarChina, {
    style: style
  }),
  countryCode: '+853',
  country: 'Macau SAR China',
  isoCode: 'mo'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Macedonia, {
    style: style
  }),
  countryCode: '+389',
  country: 'Macedonia',
  isoCode: 'mk'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Madagascar, {
    style: style
  }),
  countryCode: '+261',
  country: 'Madagascar',
  isoCode: 'mg'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Malawi, {
    style: style
  }),
  countryCode: '+265',
  country: 'Malawi',
  isoCode: 'mw'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Malaysia, {
    style: style
  }),
  countryCode: '+60',
  country: 'Malaysia',
  isoCode: 'my'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Maldives, {
    style: style
  }),
  countryCode: '+960',
  country: 'Maldives',
  isoCode: 'mv'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Mali, {
    style: style
  }),
  countryCode: '+223',
  country: 'Mali',
  isoCode: 'ml'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Malta, {
    style: style
  }),
  countryCode: '+356',
  country: 'Malta',
  isoCode: 'mt'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.MarshallIsland, {
    style: style
  }),
  countryCode: '+692',
  country: 'Marshall Islands',
  isoCode: 'mh'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Martinique, {
    style: style
  }),
  countryCode: '+596',
  country: 'Martinique',
  isoCode: 'mq'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Mauritania, {
    style: style
  }),
  countryCode: '+222',
  country: 'Mauritania',
  isoCode: 'mr'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Mauritius, {
    style: style
  }),
  countryCode: '+230',
  country: 'Mauritius',
  isoCode: 'mu'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Mayotte, {
    style: style
  }),
  countryCode: '+262',
  country: 'Mayotte',
  isoCode: 'yt'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Mexico, {
    style: style
  }),
  countryCode: '+52',
  country: 'Mexico',
  isoCode: 'mx'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Micronesia, {
    style: style
  }),
  countryCode: '+691',
  country: 'Micronesia',
  isoCode: 'fm'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Moldova, {
    style: style
  }),
  countryCode: '+373',
  country: 'Moldova',
  isoCode: 'md'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Monaco, {
    style: style
  }),
  countryCode: '+377',
  country: 'Monaco',
  isoCode: 'mc'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Mongolia, {
    style: style
  }),
  countryCode: '+976',
  country: 'Mongolia',
  isoCode: 'mn'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Montenegro, {
    style: style
  }),
  countryCode: '+382',
  country: 'Montenegro',
  isoCode: 'me'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Montserrat, {
    style: style
  }),
  countryCode: '+1',
  country: 'Montserrat',
  isoCode: 'ms'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Morocco, {
    style: style
  }),
  countryCode: '+212',
  country: 'Morocco',
  isoCode: 'ma'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Mozambique, {
    style: style
  }),
  countryCode: '+258',
  country: 'Mozambique',
  isoCode: 'mz'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Myanmar, {
    style: style
  }),
  countryCode: '+95',
  country: 'Myanmar',
  isoCode: 'mm'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Namibia, {
    style: style
  }),
  countryCode: '+264',
  country: 'Namibia',
  isoCode: 'na'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Nauru, {
    style: style
  }),
  countryCode: '+674',
  country: 'Nauru',
  isoCode: 'nr'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Nepal, {
    style: style
  }),
  countryCode: '+977',
  country: 'Nepal',
  isoCode: 'np'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Netherlands, {
    style: style
  }),
  countryCode: '+31',
  country: 'Netherlands',
  isoCode: 'nl'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.NewCaledonia, {
    style: style
  }),
  countryCode: '+687',
  country: 'New Caledonia',
  isoCode: 'nc'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.NewZealand, {
    style: style
  }),
  countryCode: '+64',
  country: 'New Zealand',
  isoCode: 'nz'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Nicaragua, {
    style: style
  }),
  countryCode: '+505',
  country: 'Nicaragua',
  isoCode: 'ni'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Niger, {
    style: style
  }),
  countryCode: '+227',
  country: 'Niger',
  isoCode: 'ne'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Nigeria, {
    style: style
  }),
  countryCode: '+234',
  country: 'Nigeria',
  isoCode: 'ng'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Niue, {
    style: style
  }),
  countryCode: '+683',
  country: 'Niue',
  isoCode: 'nu'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.NorfolkIsland, {
    style: style
  }),
  countryCode: '+672',
  country: 'Norfolk Island',
  isoCode: 'nf'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.NorthKorea, {
    style: style
  }),
  countryCode: '+850',
  country: 'North Korea',
  isoCode: 'kp'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.NorthernMarianasIslands, {
    style: style
  }),
  countryCode: '+1',
  country: 'Northern Mariana Islands',
  isoCode: 'mp'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Norway, {
    style: style
  }),
  countryCode: '+47',
  country: 'Norway',
  isoCode: 'no'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Oman, {
    style: style
  }),
  countryCode: '+968',
  country: 'Oman',
  isoCode: 'om'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Pakistan, {
    style: style
  }),
  countryCode: '+92',
  country: 'Pakistan',
  isoCode: 'pk'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Palau, {
    style: style
  }),
  countryCode: '+680',
  country: 'Palau',
  isoCode: 'pw'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Palestine, {
    style: style
  }),
  countryCode: '+970',
  country: 'Palestine',
  isoCode: 'ps'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Panama, {
    style: style
  }),
  countryCode: '+507',
  country: 'Panama',
  isoCode: 'pa'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.PapuaNewGuinea, {
    style: style
  }),
  countryCode: '+675',
  country: 'Papua New Guinea',
  isoCode: 'pg'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Paraguay, {
    style: style
  }),
  countryCode: '+595',
  country: 'Paraguay',
  isoCode: 'py'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Peru, {
    style: style
  }),
  countryCode: '+51',
  country: 'Peru',
  isoCode: 'pe'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Philippines, {
    style: style
  }),
  countryCode: '+63',
  country: 'Philippines',
  isoCode: 'ph'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Poland, {
    style: style
  }),
  countryCode: '+48',
  country: 'Poland',
  isoCode: 'pl'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Portugal, {
    style: style
  }),
  countryCode: '+351',
  country: 'Portugal',
  isoCode: 'pt'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.PuertoRico, {
    style: style
  }),
  countryCode: '+1',
  country: 'Puerto Rico',
  isoCode: 'pr'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Qatar, {
    style: style
  }),
  countryCode: '+974',
  country: 'Qatar',
  isoCode: 'qa'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Reunion, {
    style: style
  }),
  countryCode: '+262',
  country: 'Réunion',
  isoCode: 're'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Romania, {
    style: style
  }),
  countryCode: '+40',
  country: 'Romania',
  isoCode: 'ro'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Russia, {
    style: style
  }),
  countryCode: '+7',
  country: 'Russia',
  isoCode: 'ru'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Rwanda, {
    style: style
  }),
  countryCode: '+250',
  country: 'Rwanda',
  isoCode: 'rw'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.SaintBarthelemy, {
    style: style
  }),
  countryCode: '+590',
  country: 'Saint Barthélemy',
  isoCode: 'bl'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.SaintHelena, {
    style: style
  }),
  countryCode: '+290',
  country: 'Saint Helena',
  isoCode: 'sh'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.SaintKittsAndNevis, {
    style: style
  }),
  countryCode: '+1',
  country: 'Saint Kitts and Nevis',
  isoCode: 'kn'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.SaintLucia, {
    style: style
  }),
  countryCode: '+1',
  country: 'Saint Lucia',
  isoCode: 'lc'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.SaintMartin, {
    style: style
  }),
  countryCode: '+590',
  country: 'Saint Martin',
  isoCode: 'mf'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.SaintPierreAndMiquelon, {
    style: style
  }),
  countryCode: '+508',
  country: 'Saint Pierre and Miquelon',
  isoCode: 'pm'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.SaintVincentAndTheGrenadines, {
    style: style
  }),
  countryCode: '+1',
  country: 'Saint Vincent and the Grenadines',
  isoCode: 'vc'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Samoa, {
    style: style
  }),
  countryCode: '+685',
  country: 'Samoa',
  isoCode: 'ws'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.SanMarino, {
    style: style
  }),
  countryCode: '+378',
  country: 'San Marino',
  isoCode: 'sm'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.SaoTomeAndPrince, {
    style: style
  }),
  countryCode: '+239',
  country: 'São Tomé and Príncipe',
  isoCode: 'st'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.SaudiArabia, {
    style: style
  }),
  countryCode: '+966',
  country: 'Saudi Arabia',
  isoCode: 'sa'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Senegal, {
    style: style
  }),
  countryCode: '+221',
  country: 'Senegal',
  isoCode: 'sn'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Serbia, {
    style: style
  }),
  countryCode: '+381',
  country: 'Serbia',
  isoCode: 'rs'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Seychelles, {
    style: style
  }),
  countryCode: '+248',
  country: 'Seychelles',
  isoCode: 'sc'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.SierraLeone, {
    style: style
  }),
  countryCode: '+232',
  country: 'Sierra Leone',
  isoCode: 'sl'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Singapore, {
    style: style
  }),
  countryCode: '+65',
  country: 'Singapore',
  isoCode: 'sg'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.SintMaarten, {
    style: style
  }),
  countryCode: '+1',
  country: 'Sint Maarten',
  isoCode: 'sx'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Slovakia, {
    style: style
  }),
  countryCode: '+421',
  country: 'Slovakia',
  isoCode: 'sk'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Slovenia, {
    style: style
  }),
  countryCode: '+386',
  country: 'Slovenia',
  isoCode: 'si'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.SolomonIslands, {
    style: style
  }),
  countryCode: '+677',
  country: 'Solomon Islands',
  isoCode: 'sb'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Somalia, {
    style: style
  }),
  countryCode: '+252',
  country: 'Somalia',
  isoCode: 'so'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.SouthAfrica, {
    style: style
  }),
  countryCode: '+27',
  country: 'South Africa',
  isoCode: 'za'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.SouthKorea, {
    style: style
  }),
  countryCode: '+82',
  country: 'South Korea',
  isoCode: 'kr'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.SouthSudan, {
    style: style
  }),
  countryCode: '+211',
  country: 'South Sudan',
  isoCode: 'ss'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Spain, {
    style: style
  }),
  countryCode: '+34',
  country: 'Spain',
  isoCode: 'es'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.SriLanka, {
    style: style
  }),
  countryCode: '+94',
  country: 'Sri Lanka',
  isoCode: 'lk'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Sudan, {
    style: style
  }),
  countryCode: '+249',
  country: 'Sudan',
  isoCode: 'sd'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Suriname, {
    style: style
  }),
  countryCode: '+597',
  country: 'Suriname',
  isoCode: 'sr'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.SvalbardAndJanMayen, {
    style: style
  }),
  countryCode: '+47',
  country: 'Svalbard and Jan Mayen',
  isoCode: 'sj'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Swaziland, {
    style: style
  }),
  countryCode: '+268',
  country: 'Swaziland',
  isoCode: 'sz'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Sweden, {
    style: style
  }),
  countryCode: '+46',
  country: 'Sweden',
  isoCode: 'se'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Switzerland, {
    style: style
  }),
  countryCode: '+41',
  country: 'Switzerland',
  isoCode: 'ch'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Syria, {
    style: style
  }),
  countryCode: '+963',
  country: 'Syria',
  isoCode: 'sy'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Taiwan, {
    style: style
  }),
  countryCode: '+886',
  country: 'Taiwan',
  isoCode: 'tw'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Tajikistan, {
    style: style
  }),
  countryCode: '+992',
  country: 'Tajikistan',
  isoCode: 'tj'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Tanzania, {
    style: style
  }),
  countryCode: '+255',
  country: 'Tanzania',
  isoCode: 'tz'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Thailand, {
    style: style
  }),
  countryCode: '+66',
  country: 'Thailand',
  isoCode: 'th'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.TimorLeste, {
    style: style
  }),
  countryCode: '+670',
  country: 'Timor-Leste',
  isoCode: 'tl'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Togo, {
    style: style
  }),
  countryCode: '+228',
  country: 'Togo',
  isoCode: 'tg'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Tokelau, {
    style: style
  }),
  countryCode: '+690',
  country: 'Tokelau',
  isoCode: 'tk'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Tonga, {
    style: style
  }),
  countryCode: '+676',
  country: 'Tonga',
  isoCode: 'to'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.TrinidadAndTobago, {
    style: style
  }),
  countryCode: '+1',
  country: 'Trinidad and Tobago',
  isoCode: 'tt'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Tunisia, {
    style: style
  }),
  countryCode: '+216',
  country: 'Tunisia',
  isoCode: 'tn'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Turkey, {
    style: style
  }),
  countryCode: '+90',
  country: 'Turkey',
  isoCode: 'tr'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Turkmenistan, {
    style: style
  }),
  countryCode: '+993',
  country: 'Turkmenistan',
  isoCode: 'tm'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.TurksAndCaicos, {
    style: style
  }),
  countryCode: '+1',
  country: 'Turks and Caicos Islands',
  isoCode: 'tc'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Tuvalu, {
    style: style
  }),
  countryCode: '+688',
  country: 'Tuvalu',
  isoCode: 'tv'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.UsVirginIslands, {
    style: style
  }),
  countryCode: '+1',
  country: 'U.S Virgin Islands',
  isoCode: 'vi'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Uganda, {
    style: style
  }),
  countryCode: '+256',
  country: 'Uganda',
  isoCode: 'ug'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Ukraine, {
    style: style
  }),
  countryCode: '+380',
  country: 'Ukraine',
  isoCode: 'ua'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.UnitedArabEmirates, {
    style: style
  }),
  countryCode: '+971',
  country: 'United Arab Emirates',
  isoCode: 'ae'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.UnitedKingdom, {
    style: style
  }),
  countryCode: '+44',
  country: 'United Kingdom',
  isoCode: 'gb'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.UnitedStates, {
    style: style
  }),
  countryCode: '+1',
  country: 'United States',
  isoCode: 'us'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Uruguay, {
    style: style
  }),
  countryCode: '+598',
  country: 'Uruguay',
  isoCode: 'uy'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Uzbekistan, {
    style: style
  }),
  countryCode: '+998',
  country: 'Uzbekistan',
  isoCode: 'uz'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Vanuatu, {
    style: style
  }),
  countryCode: '+678',
  country: 'Vanuatu',
  isoCode: 'vu'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.VaticanCity, {
    style: style
  }),
  countryCode: '+379',
  country: 'Vatican City',
  isoCode: 'va'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Venezuela, {
    style: style
  }),
  countryCode: '+58',
  country: 'Venezuela',
  isoCode: 've'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Vietnam, {
    style: style
  }),
  countryCode: '+84',
  country: 'Vietnam',
  isoCode: 'vn'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.WallisAndFutuna, {
    style: style
  }),
  countryCode: '+681',
  country: 'Wallis and Futuna',
  isoCode: 'wf'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Yemen, {
    style: style
  }),
  countryCode: '+967',
  country: 'Yemen',
  isoCode: 'ye'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Zambia, {
    style: style
  }),
  countryCode: '+260',
  country: 'Zambia',
  isoCode: 'zm'
}, {
  flagIcon: /*#__PURE__*/_react["default"].createElement(Flags.Zimbabwe, {
    style: style
  }),
  countryCode: '+263',
  country: 'Zimbabwe',
  isoCode: 'zw'
}];
exports["default"] = _default;
