// react-rainbow-components/components/PhoneInput/index.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _hooks = require("@rainbow-modules/hooks");

var _label = _interopRequireDefault(require("../Input/label"));

var _RenderIf = _interopRequireDefault(require("../RenderIf"));

var _container = _interopRequireDefault(require("../Input/styled/container"));

var _helpText = _interopRequireDefault(require("../Input/styled/helpText"));

var _errorText = _interopRequireDefault(require("../Input/styled/errorText"));

var _AssistiveText = _interopRequireDefault(require("../AssistiveText"));

var _styled = require("./styled");

var _hooks2 = require("../../libs/hooks");

var _hooks3 = require("./hooks");

var _countriesDropdown = _interopRequireDefault(require("./countriesDropdown"));

/* eslint-disable react/no-unused-prop-types */

/**
 * phone input are used for freeform data entry.
 * @category Form
 */
var PhoneInput = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  var _useReduxForm = (0, _hooks2.useReduxForm)(props),
      value = _useReduxForm.value,
      name = _useReduxForm.name,
      placeholder = _useReduxForm.placeholder,
      icon = _useReduxForm.icon,
      bottomHelpText = _useReduxForm.bottomHelpText,
      required = _useReduxForm.required,
      error = _useReduxForm.error,
      disabled = _useReduxForm.disabled,
      readOnly = _useReduxForm.readOnly,
      tabIndex = _useReduxForm.tabIndex,
      onChange = _useReduxForm.onChange,
      onClick = _useReduxForm.onClick,
      onFocus = _useReduxForm.onFocus,
      onBlur = _useReduxForm.onBlur,
      className = _useReduxForm.className,
      style = _useReduxForm.style,
      id = _useReduxForm.id,
      label = _useReduxForm.label,
      labelAlignment = _useReduxForm.labelAlignment,
      hideLabel = _useReduxForm.hideLabel,
      countriesProps = _useReduxForm.countries;

  var containerRef = (0, _react.useRef)();
  var triggerRef = (0, _react.useRef)();
  var searchRef = (0, _react.useRef)();
  var inputRef = (0, _react.useRef)();
  (0, _react.useImperativeHandle)(ref, function () {
    return {
      focus: function focus() {
        inputRef.current.focus();
      },
      click: function click() {
        inputRef.current.click();
      },
      blur: function blur() {
        inputRef.current.blur();
      }
    };
  });
  var inputId = (0, _hooks2.useUniqueIdentifier)('phone-input');
  var errorMessageId = (0, _hooks2.useErrorMessageId)(error);
  var labelId = (0, _hooks2.useLabelId)(label);
  var phone = value && value.phone || '';
  var countries = (0, _hooks3.useCountries)(countriesProps);
  var country = (0, _hooks3.useCountry)(value, countries);
  var countryCode = country.countryCode,
      isoCode = country.isoCode,
      flagIcon = country.flagIcon;

  var _useFocusIndex = (0, _hooks3.useFocusIndex)(containerRef, triggerRef, searchRef, inputRef),
      _useFocusIndex2 = (0, _slicedToArray2["default"])(_useFocusIndex, 2),
      focusIndex = _useFocusIndex2[0],
      setFocusIndex = _useFocusIndex2[1];

  (0, _hooks.useOutsideClick)(containerRef, function () {
    setFocusIndex(-1);
  }, focusIndex > -1);
  var handleFocus = (0, _hooks3.useHandleFocus)(focusIndex, onFocus, setFocusIndex, value);
  var handleBlur = (0, _hooks3.useHandleBlur)(focusIndex, onBlur, value);
  var isOpen = focusIndex === 1;
  var handleCountryChange = (0, _hooks3.useHandleCountryChange)(phone, onChange, setFocusIndex, isOpen);

  function handlePhoneChange(event) {
    var rawPhone = event.target.value;
    var newPhone = rawPhone.replace(/\D/g, '');
    onChange({
      countryCode: countryCode,
      isoCode: isoCode,
      phone: newPhone
    });
  }

  function handleClick() {
    if (focusIndex === 1) {
      setFocusIndex(0);
    } else {
      setFocusIndex(1);
    }
  }

  var isFocus = focusIndex > -1;
  var formattedCountryCode = "(".concat(countryCode, ")");
  var isReadOnly = readOnly && !disabled;
  var hasTrigger = !isReadOnly && countries.length > 1;
  var isOnlyCountry = !isReadOnly && countries.length === 1;
  return /*#__PURE__*/_react["default"].createElement(_container["default"], {
    id: id,
    ref: containerRef,
    className: className,
    style: style
  }, /*#__PURE__*/_react["default"].createElement(_label["default"], {
    label: label,
    labelAlignment: labelAlignment,
    hideLabel: hideLabel,
    required: required,
    inputId: inputId,
    readOnly: readOnly,
    id: labelId
  }), /*#__PURE__*/_react["default"].createElement(_styled.StyledInputContainer, {
    disabled: disabled,
    readOnly: readOnly,
    iconPosition: "right",
    icon: icon,
    error: error,
    isFocus: isFocus
  }, /*#__PURE__*/_react["default"].createElement(_RenderIf["default"], {
    isTrue: isReadOnly
  }, /*#__PURE__*/_react["default"].createElement(_styled.StyledFlagContainer, {
    readOnly: true
  }, flagIcon)), /*#__PURE__*/_react["default"].createElement(_RenderIf["default"], {
    isTrue: hasTrigger
  }, /*#__PURE__*/_react["default"].createElement(_styled.StyledTrigger, {
    ref: triggerRef,
    onClick: handleClick,
    onFocus: function onFocus(event) {
      return handleFocus(event, 0);
    },
    onBlur: handleBlur,
    tabIndex: tabIndex,
    disabled: disabled,
    type: "button"
  }, /*#__PURE__*/_react["default"].createElement(_styled.StyledFlagContainer, {
    disabled: disabled
  }, flagIcon, /*#__PURE__*/_react["default"].createElement(_styled.StyledIndicator, {
    error: error,
    disabled: disabled
  })), /*#__PURE__*/_react["default"].createElement(_AssistiveText["default"], {
    text: "select country"
  }))), /*#__PURE__*/_react["default"].createElement(_RenderIf["default"], {
    isTrue: isOnlyCountry
  }, /*#__PURE__*/_react["default"].createElement(_styled.StyledTrigger, {
    ref: triggerRef,
    onFocus: function onFocus(event) {
      return handleFocus(event, 0);
    },
    onBlur: handleBlur,
    tabIndex: tabIndex,
    disabled: disabled
  }, /*#__PURE__*/_react["default"].createElement(_styled.StyledFlagContainer, {
    disabled: disabled
  }, flagIcon))), /*#__PURE__*/_react["default"].createElement(_styled.StyledCountryCode, null, formattedCountryCode), /*#__PURE__*/_react["default"].createElement(_styled.StyledInput, {
    id: inputId,
    ref: inputRef,
    name: name,
    value: phone,
    type: "tel",
    placeholder: placeholder,
    tabIndex: tabIndex,
    onFocus: function onFocus(event) {
      return handleFocus(event, 2);
    },
    onBlur: handleBlur,
    onClick: onClick,
    onChange: handlePhoneChange,
    disabled: disabled,
    readOnly: readOnly,
    required: required,
    "aria-labelledby": labelId,
    "aria-describedby": errorMessageId,
    iconPosition: "right",
    icon: icon,
    error: error,
    isFocus: isFocus
  }), /*#__PURE__*/_react["default"].createElement(_RenderIf["default"], {
    isTrue: icon
  }, /*#__PURE__*/_react["default"].createElement(_styled.StyledIconContainer, {
    error: error
  }, icon)), /*#__PURE__*/_react["default"].createElement(_countriesDropdown["default"], {
    country: country,
    countries: countries,
    isOpen: isOpen,
    searchRef: searchRef,
    setFocusIndex: setFocusIndex,
    handleFocus: handleFocus,
    handleBlur: handleBlur,
    onCountryChange: handleCountryChange
  })), /*#__PURE__*/_react["default"].createElement(_RenderIf["default"], {
    isTrue: bottomHelpText
  }, /*#__PURE__*/_react["default"].createElement(_helpText["default"], {
    alignSelf: "center"
  }, bottomHelpText)), /*#__PURE__*/_react["default"].createElement(_RenderIf["default"], {
    isTrue: error
  }, /*#__PURE__*/_react["default"].createElement(_errorText["default"], {
    alignSelf: "center",
    id: errorMessageId
  }, error)));
});

PhoneInput.propTypes = {
  /** Specifies the value of an input element. */
  value: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].shape({
    countryCode: _propTypes["default"].string,
    isoCode: _propTypes["default"].string,
    phone: _propTypes["default"].string
  })]),

  /** The name of the input. */
  name: _propTypes["default"].string,

  /** Text label for the input. */
  label: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].node]),

  /** Describes the position of the PhoneInput label. Options include left, center and right.
   * This value defaults to center. */
  labelAlignment: _propTypes["default"].oneOf(['left', 'center', 'right']),

  /** A boolean to hide the PhoneInput label. */
  hideLabel: _propTypes["default"].bool,

  /** Text that is displayed when the field is empty, to prompt the user for a valid entry. */
  placeholder: _propTypes["default"].string,

  /** The icon to show if it is passed. It must be a svg icon or a font icon. */
  icon: _propTypes["default"].node,

  /** Shows the help message below the input. */
  bottomHelpText: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].node]),

  /** Specifies that an input field must be filled out before submitting the form.
   * This value defaults to false. */
  required: _propTypes["default"].bool,

  /** Specifies that an input field must be filled out before submitting the form. */
  error: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].node]),

  /** Specifies that an input element should be disabled. This value defaults to false. */
  disabled: _propTypes["default"].bool,

  /** Specifies that an input field is read-only. This value defaults to false. */
  readOnly: _propTypes["default"].bool,

  /** Specifies the tab order of an element (when the tab button is used for navigating). */
  tabIndex: _propTypes["default"].oneOfType([_propTypes["default"].number, _propTypes["default"].string]),

  /** The action triggered when a value attribute changes. */
  onChange: _propTypes["default"].func,

  /** The action triggered when the element is clicked. */
  onClick: _propTypes["default"].func,

  /** The action triggered when the element receives focus. */
  onFocus: _propTypes["default"].func,

  /** The action triggered when the element releases focus. */
  onBlur: _propTypes["default"].func,

  /** A CSS class for the outer element, in addition to the component's base classes. */
  className: _propTypes["default"].string,

  /** An object with custom style applied to the outer element. */
  style: _propTypes["default"].object,

  /** The id of the outer element. */
  id: _propTypes["default"].string,

  /** Specifies the available countries for selection. */
  countries: _propTypes["default"].array
};
PhoneInput.defaultProps = {
  id: undefined,
  name: undefined,
  placeholder: null,
  icon: undefined,
  label: undefined,
  labelAlignment: 'center',
  hideLabel: false,
  bottomHelpText: null,
  required: false,
  error: null,
  disabled: false,
  readOnly: false,
  tabIndex: undefined,
  className: undefined,
  style: undefined,
  onClick: function onClick() {},
  onFocus: function onFocus() {},
  onBlur: function onBlur() {},
  onChange: function onChange() {},
  value: {
    countryCode: '+1',
    isoCode: 'us',
    phone: ''
  },
  countries: ['af', 'ax', 'al', 'dz', 'as', 'ad', 'ao', 'ai', 'ag', 'ar', 'am', 'aw', 'ac', 'au', 'at', 'az', 'bs', 'bh', 'bd', 'bb', 'by', 'be', 'bz', 'bj', 'bm', 'bt', 'bo', 'bq', 'ba', 'bw', 'br', 'vg', 'bn', 'bg', 'bf', 'bi', 'kh', 'cm', 'ca', 'cv', 'ky', 'cf', 'td', 'cl', 'cn', 'cx', 'cc', 'co', 'km', 'cg', 'cd', 'ck', 'cr', 'ci', 'hr', 'cu', 'cw', 'cy', 'cz', 'dk', 'dj', 'dm', 'do', 'ec', 'eg', 'sv', 'gq', 'er', 'ee', 'et', 'fk', 'fo', 'fj', 'fi', 'fr', 'gf', 'pf', 'ga', 'gm', 'ge', 'de', 'gh', 'gi', 'gr', 'gl', 'gd', 'gp', 'gu', 'gt', 'gg', 'gn', 'gw', 'gy', 'ht', 'hn', 'hk', 'hu', 'is', 'in', 'id', 'ir', 'iq', 'ie', 'im', 'il', 'it', 'jm', 'jp', 'je', 'jo', 'kz', 'ke', 'ki', 'kv', 'kw', 'kg', 'la', 'lv', 'lb', 'ls', 'lr', 'ly', 'li', 'lt', 'lu', 'mo', 'mk', 'mg', 'mw', 'my', 'mv', 'ml', 'mt', 'mh', 'mq', 'mr', 'mu', 'yt', 'mx', 'fm', 'md', 'mc', 'mn', 'me', 'ms', 'ma', 'mz', 'mm', 'na', 'nr', 'np', 'nl', 'nc', 'nz', 'ni', 'ne', 'ng', 'nu', 'nf', 'kp', 'mp', 'no', 'om', 'pk', 'pw', 'ps', 'pa', 'pg', 'py', 'pe', 'ph', 'pl', 'pt', 'pr', 'qa', 're', 'ro', 'ru', 'rw', 'bl', 'sh', 'kn', 'lc', 'mf', 'pm', 'vc', 'ws', 'sm', 'st', 'sa', 'sn', 'rs', 'sc', 'sl', 'sg', 'sx', 'sk', 'si', 'sb', 'so', 'za', 'kr', 'ss', 'es', 'lk', 'sd', 'sr', 'sj', 'sz', 'se', 'ch', 'sy', 'tw', 'tj', 'tz', 'th', 'tl', 'tg', 'tk', 'to', 'tt', 'tn', 'tr', 'tm', 'tc', 'tv', 'vi', 'ug', 'ua', 'ae', 'gb', 'us', 'uy', 'uz', 'vu', 'va', 've', 'vn', 'wf', 'ye', 'zm', 'zw']
};
var _default = PhoneInput;
exports["default"] = _default;
