// @assenti/rui-components/lib/list/ListItem.js
import React from 'react';
import { Icon, Checkbox, Tooltip } from '../index';
import { strinfigyClassObject } from '../utils';

var ListItem = function ListItem(props) {
  var className = {
    item: 'rui-list-item',
    roundedActive: props.roundedActive ? 'rounded' : '',
    right: props.leftBorder ? 'left' : 'right',
    active: props.isActiveItem ? 'active' : '',
    color: props.color ? props.color : 'primary',
    hover: props.hover ? 'hover' : '',
    noDivider: props.noDivider ? '' : 'divider',
    className: props.className ? props.className : ''
  };

  var handleClick = function handleClick(e, skip) {
    if (skip) {
      props.onClick(e);
      return;
    }

    if (e.currentTarget === e.target) {
      if (props.onClick) {
        props.onClick(e);
      }
    }
  };

  return /*#__PURE__*/React.createElement("div", {
    onClick: handleClick,
    tabIndex: props.tabIndex,
    onMouseLeave: function onMouseLeave(e) {
      return props.onMouseLeave ? props.onMouseLeave(e) : {};
    },
    onMouseOver: function onMouseOver(e) {
      return props.onMouseOver ? props.onMouseOver(e) : {};
    },
    className: strinfigyClassObject(className),
    style: {
      width: props.width ? props.width : ''
    }
  }, /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("div", {
    className: "rui-list-item__left-side"
  }, /*#__PURE__*/React.createElement("div", null, props.avatar && !props.icon ? props.avatar : null, props.icon && !props.avatar && (props.tooltip ? /*#__PURE__*/React.createElement(Tooltip, {
    tooltip: props.tooltip,
    position: props.tooltipPosition
  }, /*#__PURE__*/React.createElement(Icon, {
    name: props.icon,
    onClick: function onClick(e) {
      return handleClick(e, true);
    }
  })) : /*#__PURE__*/React.createElement(Icon, {
    name: props.icon,
    onClick: function onClick(e) {
      return handleClick(e, true);
    }
  })), /*#__PURE__*/React.createElement("div", {
    onClick: handleClick
  }, /*#__PURE__*/React.createElement("div", {
    className: "rui-list-item__title-row"
  }, props.checkbox ? /*#__PURE__*/React.createElement(Checkbox, {
    fill: true,
    color: props.color ? props.color : 'primary',
    className: "mr-10",
    checked: props.isActiveItem,
    onChange: handleClick
  }) : '', props.href ? /*#__PURE__*/React.createElement("a", {
    href: props.href,
    rel: "noopener noreferrer",
    target: props.blank ? '_blank' : undefined
  }, props.item) : /*#__PURE__*/React.createElement("span", {
    onClick: handleClick
  }, props.item)), !!props.subTitle && /*#__PURE__*/React.createElement("div", {
    className: "rui-list-subtitle"
  }, props.subTitle)))), /*#__PURE__*/React.createElement("div", {
    className: "rui-list-item__right-side"
  }, props.controls), props.children));
};

export default ListItem;
