// @assenti/rui-components/lib/popover/index.js
import _slicedToArray from "@babel/runtime/helpers/esm/slicedToArray";
import React, { useState, useRef } from 'react';
import { CSSTransition } from 'react-transition-group';
import { Icon } from '../index';
import { strinfigyClassObject } from '../utils';

var PopOver = function PopOver(props) {
  var _useState = useState(false),
      _useState2 = _slicedToArray(_useState, 2),
      visible = _useState2[0],
      setVisible = _useState2[1];

  var popup = useRef();
  var content = useRef();
  var className = {
    name: 'rui-popover',
    dark: props.dark ? 'dark' : '',
    position: props.position ? props.position : 'top',
    className: props.className ? props.className : ''
  };

  var getPosition = function getPosition() {
    switch (props.position) {
      case 'left':
        return 'popleft';

      case 'right':
        return 'popright';

      case 'bottom':
        return 'popdown';

      default:
        return 'popup';
    }
  };

  return /*#__PURE__*/React.createElement("div", {
    className: strinfigyClassObject(className),
    tabIndex: -1,
    ref: popup,
    onBlur: function onBlur() {
      return props.control && !props.noBlur ? props.onClose() : setVisible(false);
    },
    onMouseLeave: function onMouseLeave() {
      return props.trigger === 'hover' ? setVisible(false) : {};
    }
  }, /*#__PURE__*/React.createElement(CSSTransition, {
    in: props.control ? props.visible : visible,
    timeout: 150,
    classNames: getPosition(),
    unmountOnExit: true
  }, /*#__PURE__*/React.createElement("div", {
    className: "rui-popover__content",
    ref: content
  }, props.title ? /*#__PURE__*/React.createElement("div", {
    className: "rui-popover__content-header"
  }, /*#__PURE__*/React.createElement("span", null, props.title), /*#__PURE__*/React.createElement(Icon, {
    size: 18,
    name: "close",
    color: props.dark ? '#fff' : 'gray',
    onClick: function onClick() {
      return props.control ? props.onClose() : setVisible(false);
    }
  })) : '', /*#__PURE__*/React.createElement("div", {
    className: "rui-popover__content-body"
  }, props.content))), /*#__PURE__*/React.createElement("div", {
    className: "rui-popover__children",
    onMouseOver: function onMouseOver() {
      return props.trigger === 'hover' ? setVisible(true) : {};
    },
    onClick: function onClick() {
      return props.trigger !== 'hover' ? props.control ? {} : setVisible(!visible) : {};
    }
  }, props.children));
};

export default PopOver;
