// @assenti/rui-components/lib/breadcrumbs/index.js
import React from 'react';
import Icon from '../icon';
import { strinfigyClassObject } from '../utils';

var Breadcrumbs = function Breadcrumbs(props) {
  var className = {
    name: 'rui-breadcrumbs',
    size: props.size ? props.size : '',
    dark: props.dark ? 'dark' : '',
    color: props.color ? props.color : 'info',
    position: props.position ? props.position : '',
    className: props.className ? props.className : ''
  };
  return /*#__PURE__*/React.createElement("div", {
    className: strinfigyClassObject(className)
  }, props.crumbs.map(function (item, index) {
    return /*#__PURE__*/React.createElement("div", {
      key: index,
      className: props.isActive ? props.isActive(item, index) ? 'rui-breadcrumbs__item active' : 'rui-breadcrumbs__item' : 'rui-breadcrumbs__item',
      onClick: function onClick() {
        return props.onChange ? props.onChange(item, index) : {};
      }
    }, /*#__PURE__*/React.createElement("a", {
      href: props.urlKey ? item[props.urlKey] : ''
    }, props.icon ? /*#__PURE__*/React.createElement(Icon, {
      name: item[props.icon]
    }) : '', props.itemKey ? item[props.itemKey] : item), /*#__PURE__*/React.createElement("span", null, props.delimiter ? props.delimiter : '/'));
  }));
};

export default Breadcrumbs;
