// react-rainbow-components/components/CounterInput/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _hooks = require("../../libs/hooks");

var _RenderIf = _interopRequireDefault(require("../RenderIf"));

var _relativeElement = _interopRequireDefault(require("../Structural/relativeElement"));

var _label = _interopRequireDefault(require("../Input/label"));

var _helpText = _interopRequireDefault(require("../Input/styled/helpText"));

var _errorText = _interopRequireDefault(require("../Input/styled/errorText"));

var _withReduxForm = _interopRequireDefault(require("../../libs/hocs/withReduxForm"));

var _plus = _interopRequireDefault(require("./icons/plus"));

var _minus = _interopRequireDefault(require("./icons/minus"));

var _getValue = _interopRequireDefault(require("./helpers/getValue"));

var _getNormalizedValue = _interopRequireDefault(require("./helpers/getNormalizedValue"));

var _isButtonDisabled = _interopRequireDefault(require("./helpers/isButtonDisabled"));

var _isMax = _interopRequireDefault(require("./helpers/isMax"));

var _isMin = _interopRequireDefault(require("./helpers/isMin"));

var _styled = require("./styled");

/* eslint-disable react/no-unused-prop-types */

/**
 * CounterInput is a component that lets you enter a number. You can increase and decrease the  value using your mouse or by simply typing the number.
 * @category Form
 */
var CounterInput = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  var _useReduxForm = (0, _hooks.useReduxForm)(props),
      id = _useReduxForm.id,
      name = _useReduxForm.name,
      value = _useReduxForm.value,
      placeholder = _useReduxForm.placeholder,
      _onChange = _useReduxForm.onChange,
      tabIndex = _useReduxForm.tabIndex,
      _onFocus = _useReduxForm.onFocus,
      _onBlur = _useReduxForm.onBlur,
      onClick = _useReduxForm.onClick,
      onKeyDown = _useReduxForm.onKeyDown,
      disabled = _useReduxForm.disabled,
      readOnly = _useReduxForm.readOnly,
      required = _useReduxForm.required,
      min = _useReduxForm.min,
      max = _useReduxForm.max,
      isBare = _useReduxForm.isBare,
      error = _useReduxForm.error,
      variant = _useReduxForm.variant,
      step = _useReduxForm.step,
      label = _useReduxForm.label,
      labelAlignment = _useReduxForm.labelAlignment,
      hideLabel = _useReduxForm.hideLabel,
      bottomHelpText = _useReduxForm.bottomHelpText,
      className = _useReduxForm.className,
      style = _useReduxForm.style;

  var inputRef = (0, _react.useRef)(null);
  var inputId = (0, _hooks.useUniqueIdentifier)('counter-input');
  var errorMessageId = (0, _hooks.useErrorMessageId)(error);
  var labelId = (0, _hooks.useLabelId)(label);
  var isReadOnly = !!(!disabled && readOnly);
  (0, _react.useImperativeHandle)(ref, function () {
    return {
      focus: function focus() {
        inputRef.current.focus();
      },
      click: function click() {
        inputRef.current.click();
      },
      blur: function blur() {
        inputRef.current.blur();
      }
    };
  });

  var handlePlusMouseDown = function handlePlusMouseDown(event) {
    event.preventDefault();
    inputRef.current.focus();

    _onChange((0, _getNormalizedValue["default"])((0, _getValue["default"])(Number(value)) + step));
  };

  var handleMinusMouseDown = function handleMinusMouseDown(event) {
    event.preventDefault();
    inputRef.current.focus();

    _onChange((0, _getNormalizedValue["default"])((0, _getValue["default"])(Number(value)) - step));
  };

  var handleEvents = function handleEvents(event, callback) {
    if (event.target.value === '') {
      return callback();
    }

    return callback(Number(event.target.value));
  };

  return /*#__PURE__*/_react["default"].createElement(_styled.StyledContainer, {
    id: id,
    className: className,
    style: style
  }, /*#__PURE__*/_react["default"].createElement(_label["default"], {
    label: label,
    labelAlignment: labelAlignment,
    hideLabel: hideLabel,
    required: required,
    inputId: inputId,
    readOnly: isReadOnly,
    id: labelId
  }), /*#__PURE__*/_react["default"].createElement(_relativeElement["default"], null, /*#__PURE__*/_react["default"].createElement(_styled.ButtonContainer, {
    iconPosition: "left",
    readOnly: readOnly,
    error: error
  }, /*#__PURE__*/_react["default"].createElement(_styled.StyledButton, {
    variant: "base",
    size: "small",
    icon: /*#__PURE__*/_react["default"].createElement(_minus["default"], null),
    onMouseDown: handleMinusMouseDown,
    tabIndex: -1,
    disabled: (0, _isButtonDisabled["default"])((0, _isMin["default"])(value, step, min), disabled, readOnly),
    assistiveText: "minus button"
  })), /*#__PURE__*/_react["default"].createElement(_styled.StyledInput, {
    id: inputId,
    name: name,
    type: "number",
    value: value,
    placeholder: placeholder,
    onChange: function onChange(event) {
      return handleEvents(event, _onChange);
    },
    tabIndex: tabIndex,
    onFocus: function onFocus(event) {
      return handleEvents(event, _onFocus);
    },
    onBlur: function onBlur(event) {
      return handleEvents(event, _onBlur);
    },
    onClick: onClick,
    onKeyDown: onKeyDown,
    disabled: disabled,
    readOnly: readOnly,
    required: required,
    min: min,
    max: max,
    "aria-labelledby": labelId,
    "aria-describedby": errorMessageId,
    ref: inputRef,
    isBare: isBare,
    error: error,
    variant: variant,
    step: step,
    autoComplete: "off"
  }), /*#__PURE__*/_react["default"].createElement(_styled.ButtonContainer, {
    iconPosition: "right",
    readOnly: readOnly,
    error: error
  }, /*#__PURE__*/_react["default"].createElement(_styled.StyledButton, {
    variant: "base",
    size: "small",
    icon: /*#__PURE__*/_react["default"].createElement(_plus["default"], null),
    onMouseDown: handlePlusMouseDown,
    disabled: (0, _isButtonDisabled["default"])((0, _isMax["default"])(value, step, max), disabled, readOnly),
    tabIndex: -1,
    assistiveText: "plus button"
  }))), /*#__PURE__*/_react["default"].createElement(_RenderIf["default"], {
    isTrue: bottomHelpText
  }, /*#__PURE__*/_react["default"].createElement(_helpText["default"], {
    alignSelf: "center"
  }, bottomHelpText)), /*#__PURE__*/_react["default"].createElement(_RenderIf["default"], {
    isTrue: error
  }, /*#__PURE__*/_react["default"].createElement(_errorText["default"], {
    alignSelf: "center",
    id: errorMessageId
  }, error)));
});

var _default = (0, _withReduxForm["default"])(CounterInput);

exports["default"] = _default;
CounterInput.propTypes = {
  /** The id of the outer element. */
  id: _propTypes["default"].string,

  /** The name of the input. */
  name: _propTypes["default"].string,

  /** Specifies the value of an input element. */
  value: _propTypes["default"].number,

  /** Text that is displayed when the field is empty, to prompt the user for a valid entry. */
  placeholder: _propTypes["default"].string,

  /** The action triggered when a value attribute changes. */
  onChange: _propTypes["default"].func,

  /** Specifies the tab order of an element (when the tab button is used for navigating). */
  tabIndex: _propTypes["default"].oneOfType([_propTypes["default"].number, _propTypes["default"].string]),

  /** The action triggered when the element receives focus. */
  onFocus: _propTypes["default"].func,

  /** The action triggered when the element releases focus. */
  onBlur: _propTypes["default"].func,

  /** The action triggered when the element is clicked. */
  onClick: _propTypes["default"].func,

  /** The action triggered when a key is pressed on the element. */
  onKeyDown: _propTypes["default"].func,

  /** Specifies that an input element should be disabled. This value defaults to false. */
  disabled: _propTypes["default"].bool,

  /** Specifies that an input field is read-only. This value defaults to false. */
  readOnly: _propTypes["default"].bool,

  /** Specifies that an input field must be filled out before submitting the form.
   * This value defaults to false. */
  required: _propTypes["default"].bool,

  /** Specifies the minimum value allowed in the field. */
  min: _propTypes["default"].number,

  /** Specifies the maximum value allowed in the field. */
  max: _propTypes["default"].number,

  /** Specifies the error that the input contains. */
  error: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].node]),

  /** The variant changes the appearance of the Input. Accepted variants include default
   * and shaded. This value defaults to default. */
  variant: _propTypes["default"].oneOf(['default', 'shaded', 'bare']),

  /** A number that specifies the granularity that the value must adhere to. */
  step: _propTypes["default"].number,

  /** Text label for the Input. */
  label: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].node]),

  /** This value defaults to center. */
  labelAlignment: _propTypes["default"].oneOf(['left', 'center', 'right']),

  /** A boolean to hide the Input label. */
  hideLabel: _propTypes["default"].bool,

  /** Shows the help message below the Input. */
  bottomHelpText: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].node]),

  /** A CSS class for the outer element, in addition to the component's base classes. */
  className: _propTypes["default"].string,

  /** An object with custom style applied to the outer element. */
  style: _propTypes["default"].object
};
CounterInput.defaultProps = {
  id: undefined,
  name: undefined,
  value: undefined,
  placeholder: null,
  tabIndex: undefined,
  disabled: false,
  readOnly: false,
  required: false,
  min: undefined,
  max: undefined,
  error: null,
  step: 1,
  variant: 'default',
  onChange: function onChange() {},
  onFocus: function onFocus() {},
  onBlur: function onBlur() {},
  onClick: function onClick() {},
  onKeyDown: function onKeyDown() {},
  label: undefined,
  labelAlignment: 'center',
  hideLabel: false,
  bottomHelpText: null,
  className: undefined,
  style: undefined
};
