// @assenti/rui-components/lib/progress/index.js
import React from 'react';
import { Icon } from '../index';
import { strinfigyClassObject } from '../utils';

var Progress = function Progress(props) {
  var className = {
    name: 'rui-progress__container',
    circle: props.circle ? 'circle' : '',
    active: props.active ? 'active' : '',
    dark: props.dark ? 'dark' : '',
    color: props.color ? props.color : 'primary',
    size: props.size && props.size !== 'default' ? props.size : '',
    disabled: props.disabled ? 'disabled' : '',
    className: props.className ? props.className : ''
  };

  var sizeParams = function sizeParams() {
    if (props.size && props.size !== 'default') {
      if (props.size === 'medium') return {
        radius: 50,
        stroke: 7,
        iconSize: 28
      };else if (props.size === 'large') return {
        radius: 60,
        stroke: 9,
        iconSize: 34
      };
    } else {
      return {
        radius: 40,
        stroke: 5.5,
        iconSize: 22
      };
    }
  };

  var colorParams = function colorParams() {
    if (props.iconColor) {
      return props.iconColor;
    } else {
      switch (props.color) {
        case 'info':
          return '#42a5f5';

        case 'success':
          return '#1aaa55';

        case 'error':
          return '#f44336';

        default:
          return '#1678c2';
      }
    }
  };

  var getCompletion = function getCompletion() {
    return props.complete ? props.complete : 0;
  };

  var convertToDeg = function convertToDeg(percent) {
    return 360 * percent / 100;
  };

  var Circle = function Circle() {
    var complete = props.complete;
    var radius = props.radius ? props.radius : sizeParams().radius;
    var stroke = props.stroke ? props.stroke : sizeParams().stroke;
    var normalizedRadius = radius - stroke * 2;
    var circumference = normalizedRadius * 2 * Math.PI;
    var strokeComplete = circumference - complete / 100 * circumference;
    var strokeRemain = circumference - (100 - complete) / 100 * circumference;
    var remainPercent = strokeRemain * 100 / circumference;
    return /*#__PURE__*/React.createElement("svg", {
      height: radius * 2,
      width: radius * 2
    }, /*#__PURE__*/React.createElement("circle", {
      className: "circle",
      strokeWidth: stroke,
      strokeDasharray: circumference + ' ' + circumference,
      style: {
        strokeDashoffset: strokeComplete,
        transform: "rotate(-90deg)"
      },
      r: normalizedRadius,
      cx: radius,
      cy: radius
    }), /*#__PURE__*/React.createElement("circle", {
      className: "circle-remain",
      strokeWidth: stroke,
      strokeDasharray: circumference + ' ' + circumference,
      style: {
        strokeDashoffset: strokeRemain,
        transform: "rotate(".concat(-90 + convertToDeg(remainPercent), "deg)")
      },
      r: normalizedRadius,
      cx: radius,
      cy: radius
    }));
  };

  return /*#__PURE__*/React.createElement("div", {
    className: strinfigyClassObject(className)
  }, !props.circle ? /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("div", {
    className: "rui-progress",
    style: {
      width: props.width ? props.width : 250
    }
  }, /*#__PURE__*/React.createElement("div", {
    className: "rui-progress__complete",
    style: {
      width: "".concat(getCompletion(), "%")
    }
  })), !props.hideState && !props.icon ? /*#__PURE__*/React.createElement("div", {
    className: "rui-progress__state"
  }, props.complete ? props.complete : 0, /*#__PURE__*/React.createElement("small", null, "%")) : '', props.icon ? /*#__PURE__*/React.createElement(Icon, {
    className: "ml-5",
    name: props.icon,
    color: colorParams(),
    size: sizeParams().iconSize
  }) : '') : /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Circle, null), !props.hideState && !props.icon ? /*#__PURE__*/React.createElement("div", {
    className: "rui-progress__state circle"
  }, props.complete ? props.complete : 0, /*#__PURE__*/React.createElement("small", null, "%")) : '', props.icon ? /*#__PURE__*/React.createElement(Icon, {
    className: "circle",
    name: props.icon,
    color: colorParams()
  }) : ''));
};

export default Progress;
