// @assenti/rui-components/lib/button/index.js
import React from 'react';
import { strinfigyClassObject } from '../utils';

var Button = function Button(props) {
  var className = {
    btn: 'rui-btn',
    lifted: props.lifted ? 'lifted' : '',
    loading: props.loading ? 'loading' : '',
    uppercase: props.uppercase ? 'uppercase' : '',
    borderType: props.borderType ? props.borderType : '',
    color: props.color && !props.disabled && !props.loading && !props.light && !props.dark ? props.color : 'secondary',
    light: props.light && !props.dark ? 'light' : '',
    dark: props.dark && !props.light ? 'dark' : '',
    icon: props.icon && !props.name ? 'icon' : '',
    disabled: props.disabled || props.loading ? 'disabled' : '',
    size: props.size ? props.size : '',
    block: props.block ? 'block' : '',
    outlined: props.outlined ? 'outlined' : '',
    allotted: props.iconAllotted ? 'allotted' : '',
    iconLeft: props.iconLeft ? 'icon-left' : '',
    className: props.className ? props.className : ''
  };

  var setMargin = function setMargin() {
    if (props.icon && !props.iconLeft) return 'rui-btn__text mr-5';else if (props.icon && props.iconLeft) return 'rui-btn__text ml-5';else return 'rui-btn__text';
  };

  return /*#__PURE__*/React.createElement("button", {
    type: props.type ? props.type : 'button',
    ref: props.btnRef,
    id: props.id,
    disabled: props.disabled || props.loading,
    className: strinfigyClassObject(className),
    onClick: function onClick(e) {
      return props.onClick ? props.onClick(e) : {};
    }
  }, props.loading ? /*#__PURE__*/React.createElement("div", {
    className: "rui-btn-loading"
  }, /*#__PURE__*/React.createElement("span", null, "\u2022\u2022\u2022")) : /*#__PURE__*/React.createElement(React.Fragment, null, props.icon && props.iconLeft && (props.iconAllotted ? /*#__PURE__*/React.createElement("div", {
    className: "rui-btn__allotted-icon"
  }, props.icon) : props.icon), props.name ? /*#__PURE__*/React.createElement("div", {
    className: setMargin()
  }, props.name) : '', props.icon && !props.iconLeft && (props.iconAllotted ? /*#__PURE__*/React.createElement("div", {
    className: "rui-btn__allotted-icon"
  }, props.icon) : props.icon)));
};

export default Button;
