// @assenti/rui-components/lib/buttonGroup/index.js
import _slicedToArray from "@babel/runtime/helpers/esm/slicedToArray";
import React, { useState } from 'react';
import { Button, Icon } from '../index';
import { strinfigyClassObject } from '../utils';

var ButtonGroup = function ButtonGroup(props) {
  var _useState = useState(props.default),
      _useState2 = _slicedToArray(_useState, 2),
      active = _useState2[0],
      setActive = _useState2[1];

  var className = {
    btn: 'rui-btn-group',
    lifted: props.lifted ? 'lifted' : '',
    icons: props.icon ? 'icons' : '',
    size: props.size ? props.size : '',
    className: props.className ? props.className : ''
  };

  var isActive = function isActive(index) {
    return index === active ? 'active' : '';
  };

  var handleChange = function handleChange(item, index) {
    if (props.onChange) props.onChange(item, index);
    setActive(index);
  };

  return /*#__PURE__*/React.createElement("div", {
    className: strinfigyClassObject(className)
  }, props.options.map(function (item, index) {
    return /*#__PURE__*/React.createElement(Button, {
      key: index,
      size: props.size ? props.size : '',
      className: isActive(index),
      name: !props.icon ? item : '',
      icon: props.icon ? /*#__PURE__*/React.createElement(Icon, {
        name: item
      }) : '',
      color: props.color ? props.color : '',
      onClick: function onClick() {
        return handleChange(item, index);
      },
      outlined: props.outlined ? true : false
    });
  }));
};

export default ButtonGroup;
