// @assenti/rui-components/lib/alert/index.js
import React from 'react';
import { CSSTransition } from 'react-transition-group';
import { Icon } from '../index';
import { strinfigyClassObject } from '../utils';

var Alert = function Alert(props) {
  var className = {
    name: 'rui-alert',
    color: props.status ? props.status : 'info',
    className: props.className ? props.className : ''
  };

  var getStatusIcon = function getStatusIcon() {
    switch (props.status) {
      case 'warning':
        return 'alert';

      case 'error':
        return 'alert-circle';

      case 'success':
        return 'check-circle';

      default:
        return 'info';
    }
  };

  return /*#__PURE__*/React.createElement(CSSTransition, {
    in: props.visible,
    timeout: 300,
    classNames: "rui-alert",
    unmountOnExit: true
  }, /*#__PURE__*/React.createElement("div", {
    className: strinfigyClassObject(className)
  }, /*#__PURE__*/React.createElement("div", null, /*#__PURE__*/React.createElement("div", {
    className: "rui-alert__status"
  }, /*#__PURE__*/React.createElement(Icon, {
    name: getStatusIcon(),
    size: 20
  }), props.message), /*#__PURE__*/React.createElement("div", {
    className: "rui-alert__text"
  }, props.text)), /*#__PURE__*/React.createElement(Icon, {
    className: "rui-alert__close",
    name: "close",
    size: 18,
    onClick: function onClick() {
      return props.onClose ? props.onClose() : {};
    }
  })));
};

export default Alert;
