// @assenti/rui-components/lib/drawer/index.js
import React, { useEffect, useRef } from 'react';
import { CSSTransition } from 'react-transition-group';
import { strinfigyClassObject } from '../utils';

var Drawer = function Drawer(props) {
  var drawer = useRef();
  var className = {
    name: 'rui-drawer',
    position: props.position ? props.position : 'left',
    dark: props.dark && !props.light ? 'dark' : ''
  };

  var handleClick = function handleClick(e) {
    if (drawer.current && drawer.current.contains(e.target)) return;
    props.onClose();
  };

  useEffect(function () {
    document.addEventListener('mousedown', handleClick, true);
    return function () {
      document.removeEventListener("mousedown", handleClick, true);
    };
  }, []);
  return /*#__PURE__*/React.createElement(CSSTransition, {
    in: props.drawer,
    timeout: 300,
    classNames: "rui-drawer-".concat(props.position ? props.position : 'left'),
    unmountOnExit: true
  }, /*#__PURE__*/React.createElement("div", {
    className: "rui-drawer-overlay"
  }, /*#__PURE__*/React.createElement("div", {
    ref: drawer,
    style: {
      width: props.width ? props.width : 280
    },
    className: strinfigyClassObject(className)
  }, props.children)));
};

export default Drawer;
