// @assenti/rui-components/lib/checkbox/index.js
import React from 'react';
import { strinfigyClassObject } from '../utils';

var Checkbox = function Checkbox(props) {
  var className = {
    checkbox: 'rui-checkbox',
    fill: props.fill ? 'fill' : '',
    size: props.size ? props.size : '',
    color: props.color ? props.color : 'primary',
    disabled: props.disabled ? 'disabled' : ''
  };

  var checkboxLabelClass = function checkboxLabelClass() {
    var result = '';
    var className = {
      checkbox: 'rui-checkbox-label',
      color: props.color ? props.color : '',
      disabled: props.disabled ? 'disabled' : ''
    };

    for (var key in className) {
      if (className[key]) result += className[key] + ' ';
    }

    return result.trim();
  };

  return /*#__PURE__*/React.createElement("div", {
    className: props.className ? "rui-checkbox-wrapper ".concat(props.className) : 'rui-checkbox-wrapper'
  }, /*#__PURE__*/React.createElement("label", {
    className: strinfigyClassObject(className)
  }, /*#__PURE__*/React.createElement("input", {
    type: "checkbox",
    checked: props.checked ? props.checked : false,
    disabled: props.disabled ? props.disabled : false,
    onChange: function onChange(e) {
      return props.onChange ? props.onChange(e) : {};
    }
  }), /*#__PURE__*/React.createElement("span", null)), props.label ? /*#__PURE__*/React.createElement("div", {
    className: checkboxLabelClass(),
    onClick: function onClick() {
      return props.onChange && !props.disabled ? props.onChange() : {};
    }
  }, props.label) : '');
};

export default Checkbox;
