// @assenti/rui-components/lib/tableSimple/index.js
import React from 'react';
import { strinfigyClassObject } from '../utils';

var TableSimple = function TableSimple(props) {
  var className = {
    name: 'rui-table',
    dark: props.dark ? 'dark' : '',
    noHover: props.noHover ? 'no-hover' : '',
    stripped: props.stripped ? 'stripped' : '',
    bordered: props.bordered ? 'bordered' : '',
    grid: props.grid ? 'grid' : '',
    className: props.className ? props.className : ''
  };
  return /*#__PURE__*/React.createElement("table", {
    ref: props.tableRef,
    style: props.style,
    id: props.id,
    className: strinfigyClassObject(className)
  }, props.children);
};

var TableHead = function TableHead(props) {
  return /*#__PURE__*/React.createElement("thead", {
    style: props.style,
    className: props.className
  }, props.children);
};

var TableBody = function TableBody(props) {
  return /*#__PURE__*/React.createElement("tbody", {
    style: props.style,
    className: props.className
  }, props.children);
};

var TableRow = function TableRow(props) {
  return /*#__PURE__*/React.createElement("tr", {
    style: props.style,
    className: props.className
  }, props.children);
};

var TableHeader = function TableHeader(props) {
  var className = {
    align: props.align ? props.align : '',
    className: props.className ? props.className : ''
  };
  return /*#__PURE__*/React.createElement("th", {
    style: props.style,
    className: strinfigyClassObject(className),
    colSpan: props.colSpan,
    rowSpan: props.rowSpan
  }, props.children);
};

var TableCell = function TableCell(props) {
  var className = {
    align: props.align ? props.align : '',
    className: props.className ? props.className : ''
  };
  return /*#__PURE__*/React.createElement("td", {
    style: props.style,
    className: strinfigyClassObject(className),
    colSpan: props.colSpan,
    rowSpan: props.rowSpan
  }, props.children);
};

TableSimple.Head = TableHead;
TableSimple.Body = TableBody;
TableSimple.Row = TableRow;
TableSimple.Cell = TableCell;
TableSimple.Header = TableHeader;
export default TableSimple;
