// @assenti/rui-components/lib/radio/index.js
import React from 'react';
import { strinfigyClassObject } from '../utils';

var Radio = function Radio(props) {
  var className = {
    name: 'rui-radio',
    color: props.color ? props.color : 'primary',
    size: props.size && props.size !== 'default' ? props.size : '',
    disabled: props.disabled ? 'disabled' : '',
    className: props.className ? props.className : ''
  };
  return /*#__PURE__*/React.createElement("label", {
    className: strinfigyClassObject(className)
  }, /*#__PURE__*/React.createElement("input", {
    type: "radio",
    name: props.name,
    value: props.value,
    checked: props.value,
    disabled: props.disabled,
    onChange: function onChange(e) {
      return props.onChange(e);
    }
  }), /*#__PURE__*/React.createElement("span", {
    className: "rui-radio-checkmark"
  }), props.label);
};

var RadioGroup = function RadioGroup(props) {
  var className = {
    btn: 'rui-radio-group',
    vertical: props.vertical ? 'vertical' : '',
    className: props.className ? props.className : ''
  };

  var isChecked = function isChecked() {
    for (var i = 0; i < props.options.length; i++) {
      if (props.options[i] === props.value) return i;
    }

    return -1;
  };

  return /*#__PURE__*/React.createElement("div", {
    className: strinfigyClassObject(className)
  }, props.options.map(function (item, index) {
    return /*#__PURE__*/React.createElement(Radio, {
      key: index,
      color: props.color,
      disabled: props.disabled,
      size: props.size,
      label: props.labelName ? props.labelName : item,
      value: isChecked() === index ? props.value : '',
      name: props.name,
      onChange: function onChange() {
        return props.onChange(item);
      }
    });
  }));
};

export default RadioGroup;
