// @assenti/rui-components/lib/tabs/index.js
import _slicedToArray from "@babel/runtime/helpers/esm/slicedToArray";
import React, { useState } from 'react';
import { SwitchTransition, CSSTransition } from 'react-transition-group';
import { Icon } from '../index';
import { strinfigyClassObject } from '../utils';

var Tabs = function Tabs(props) {
  var _useState = useState(props.defaultTab ? props.defaultTab : 0),
      _useState2 = _slicedToArray(_useState, 2),
      active = _useState2[0],
      setActive = _useState2[1];

  var _useState3 = useState('rui-tabs-next'),
      _useState4 = _slicedToArray(_useState3, 2),
      activeClass = _useState4[0],
      setActiveClass = _useState4[1];

  var className = {
    name: 'rui-tabs',
    pos: props.position && props.position !== 'default' ? props.position : '',
    className: props.className ? props.className : ''
  };
  var classNameHeader = {
    name: 'rui-tabs__header',
    color: props.color ? props.color : 'primary',
    size: props.size && props.size !== 'default' ? props.size : '',
    centered: props.centered ? 'center' : '',
    stretch: props.stretch ? 'stretch' : ''
  };

  var getIconColor = function getIconColor(isActive) {
    switch (props.color) {
      case 'info':
        return isActive ? '#42a5f5' : '';

      case 'success':
        return isActive ? '#1aaa55' : '';

      case 'error':
        return isActive ? '#f44336' : '';

      case 'dark':
        return isActive ? '#4d4d4d' : '';

      default:
        return isActive ? '#1678c2' : '';
    }
  };

  return /*#__PURE__*/React.createElement("div", {
    className: strinfigyClassObject(className)
  }, /*#__PURE__*/React.createElement("div", {
    className: strinfigyClassObject(classNameHeader)
  }, props.tabs.map(function (item, index) {
    return /*#__PURE__*/React.createElement("div", {
      key: index,
      onClick: function onClick() {
        setActive(index);
        if (props.onChange) props.onChange(index);
      },
      className: index === active ? "rui-tabs__tab active ".concat(props.activeHighlightFill ? 'fill' : '') : "rui-tabs__tab ".concat(props.activeHighlightFill ? 'fill' : '')
    }, props.tabIconKey ? /*#__PURE__*/React.createElement(Icon, {
      size: 18,
      className: "mr-5",
      name: item[props.tabIconKey],
      color: getIconColor(index === active)
    }) : '', props.tabTitleKey ? item[props.tabTitleKey] : item);
  })), /*#__PURE__*/React.createElement("div", {
    className: "rui-tabs__content",
    style: {
      padding: props.contentPadding
    }
  }, /*#__PURE__*/React.createElement(SwitchTransition, {
    mode: "out-in"
  }, /*#__PURE__*/React.createElement(CSSTransition, {
    key: active,
    addEndListener: function addEndListener(node, done) {
      node.addEventListener("transitionend", done, false);
    },
    classNames: activeClass
  }, props.tabItems[active]))));
};

export default Tabs;
