// react-rainbow-components/components/MonthlyCalendar/pageObject/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

/**
 * MonthlyCalendar page object class.
 * @class
 * @tutorial Monthcalendar
 */
var PageMonthlyCalendar = /*#__PURE__*/function () {
  /**
   * Create a new PageMonthlyCalendar page object.
   * @constructor
   * @param {string} rootElement - The selector of the PageMonthlyCalendar root element.
   */
  function PageMonthlyCalendar(rootElement) {
    (0, _classCallCheck2["default"])(this, PageMonthlyCalendar);
    this.rootElement = rootElement;
  }
  /**
   * Clicks the previous month button element.
   * @method
   */


  (0, _createClass2["default"])(PageMonthlyCalendar, [{
    key: "clickPrevMonthButton",
    value: function clickPrevMonthButton() {
      $(this.rootElement).$$('button[data-id=button-icon-element]')[0].click();
    }
    /**
     * Clicks the next month button element.
     * @method
     */

  }, {
    key: "clickNextMonthButton",
    value: function clickNextMonthButton() {
      $(this.rootElement).$$('button[data-id=button-icon-element]')[1].click();
    }
    /**
     * Clicks the select year element.
     * @method
     */

  }, {
    key: "clickSelectYear",
    value: function clickSelectYear() {
      $(this.rootElement).$('select').click();
    }
    /**
     * Returns the text of the current selected month element.
     * @method
     * @returns {string}
     */

  }, {
    key: "getSelectedMonth",
    value: function getSelectedMonth() {
      return $(this.rootElement).$('h3[data-id=month]').getText();
    }
    /**
     * Returns the value of the select year element.
     * @method
     * @returns {string}
     */

  }, {
    key: "getSelectedYear",
    value: function getSelectedYear() {
      return $(this.rootElement).$('select').getValue();
    }
    /**
     * Set the value of the year select element
     * @method
     * @param {string}
     */

  }, {
    key: "setYear",
    value: function setYear(value) {
      $(this.rootElement).$('select').selectByVisibleText(value);
    }
    /**
     * Returns true when the previous month button element is disabled.
     * @method
     * @returns {bool}
     */

  }, {
    key: "isPrevMonthButtonDisabled",
    value: function isPrevMonthButtonDisabled() {
      var buttonEl = $(this.rootElement).$$('button[data-id=button-icon-element]')[0];
      return !buttonEl.isEnabled();
    }
    /**
     * Returns true when the next month button element is disabled.
     * @method
     * @returns {bool}
     */

  }, {
    key: "isNextMonthButtonDisabled",
    value: function isNextMonthButtonDisabled() {
      var buttonEl = $(this.rootElement).$$('button[data-id=button-icon-element]')[1];
      return !buttonEl.isEnabled();
    }
  }]);
  return PageMonthlyCalendar;
}();

module.exports = PageMonthlyCalendar;
