// react-rainbow-components/components/PrimitiveMenu/pageObject/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var PageMenuItem = require('../../MenuItem/pageObject');
/**
 * PrimitiveMenu page object class.
 * @class
 */


var PagePrimitiveMenu = /*#__PURE__*/function () {
  /**
   * Create a new PrimitiveMenu page object.
   * @constructor
   * @param {string} rootElement - The selector of the PrimitiveMenu root element.
   */
  function PagePrimitiveMenu(triggerElement, dropdownElement) {
    (0, _classCallCheck2["default"])(this, PagePrimitiveMenu);
    this.triggerElement = triggerElement;
    this.dropdownElement = dropdownElement || 'div[data-id="primitive-menu_dropdown"]';
  }
  /**
   * Clicks the PrimitiveMenu trigger element.
   * @method
   */


  (0, _createClass2["default"])(PagePrimitiveMenu, [{
    key: "clickTrigger",
    value: function clickTrigger() {
      $(this.triggerElement).click();
    }
    /**
     * Returns true when the menu is open, false otherwise.
     * @method
     * @returns {bool}
     */

  }, {
    key: "isDropdownOpen",
    value: function isDropdownOpen() {
      return $(this.dropdownElement).isExisting() && $(this.dropdownElement).isDisplayed();
    }
    /**
     * Returns true when the button element has focus.
     * @method
     * @returns {bool}
     */

  }, {
    key: "hasFocusTrigger",
    value: function hasFocusTrigger() {
      return $(this.triggerElement).isFocused();
    }
    /**
     * @method
     * @param {number} index
     * @return {PageMenuItem}
     */

  }, {
    key: "getItem",
    value: function getItem(itemPosition) {
      var menuItems = $(this.dropdownElement).$$('li[role="menuitem"]');

      if (menuItems[itemPosition]) {
        return new PageMenuItem(menuItems[itemPosition]);
      }

      return null;
    }
  }]);
  return PagePrimitiveMenu;
}();

module.exports = PagePrimitiveMenu;
