// @assenti/rui-components/lib/timeline/index.js
import React from 'react';
import { Tag, Card, Icon } from '../index';
import { strinfigyClassObject } from '../utils';

var Timeline = function Timeline(props) {
  var className = {
    name: 'rui-timeline',
    centered: props.centered ? 'centered' : '',
    color: props.color ? props.color : 'primary',
    className: props.className ? props.className : ''
  };

  var getItemClass = function getItemClass(index) {
    if (props.shuffled) {
      if (index % 2 === 0) {
        return 'rui-timeline__item shuffled';
      } else {
        return 'rui-timeline__item';
      }
    } else {
      return 'rui-timeline__item';
    }
  };

  return /*#__PURE__*/React.createElement("div", {
    className: strinfigyClassObject(className)
  }, props.items.map(function (item, index) {
    return /*#__PURE__*/React.createElement("div", {
      key: index,
      className: getItemClass(index)
    }, /*#__PURE__*/React.createElement("div", {
      className: "rui-timeline__item__event"
    }, props.date && /*#__PURE__*/React.createElement(Tag, {
      small: true,
      outlined: props.tagOutlined,
      borderType: props.tagBorderType,
      color: props.tagColor ? props.tagColor : props.color
    }, item[props.date])), /*#__PURE__*/React.createElement("div", {
      className: "rui-timeline__item__center"
    }, /*#__PURE__*/React.createElement("div", {
      className: "rui-timeline__item-line"
    }), /*#__PURE__*/React.createElement("div", {
      className: props.icon ? 'rui-timeline__item-point icon' : 'rui-timeline__item-point'
    }, props.icon ? /*#__PURE__*/React.createElement(Icon, {
      name: props.icon,
      color: props.color === 'yellow' ? '' : '#fff',
      size: 16
    }) : null)), !props.data ? /*#__PURE__*/React.createElement(Card, {
      flat: props.flatCard
    }, !!props.title && /*#__PURE__*/React.createElement("div", {
      className: "rui-timeline__title"
    }, item[props.title]), !!props.subtitle && /*#__PURE__*/React.createElement("div", {
      className: "rui-timeline__subtitle"
    }, item[props.subtitle])) : props.data);
  }));
};

export default Timeline;
