// @assenti/rui-components/lib/avatar/index.js
import React from 'react';
import { Icon } from '../index';
import { strinfigyClassObject } from '../utils';

var Avatar = function Avatar(props) {
  var className = {
    name: 'rui-avatar',
    dark: props.dark ? 'dark' : '',
    lifted: props.lifted ? 'lifted' : '',
    transparent: props.transparent ? 'transparent' : '',
    borderType: props.borderType ? props.borderType : '',
    className: props.className ? props.className : ''
  };
  return /*#__PURE__*/React.createElement("div", {
    ref: props.avatarRef,
    className: strinfigyClassObject(className),
    style: {
      width: props.width ? props.width : 100,
      minWidth: props.width ? props.width : 100,
      height: props.height ? props.height : 120
    },
    onClick: function onClick(e) {
      return props.onClick ? props.onClick(e) : {};
    }
  }, props.img && !props.icon && !props.initials ? /*#__PURE__*/React.createElement("img", {
    src: props.img,
    alt: ""
  }) : null, props.initials && !props.icon && !props.img ? /*#__PURE__*/React.createElement("span", {
    style: {
      fontSize: props.fontSize
    }
  }, props.initials) : null, props.icon && !props.img && !props.initials ? /*#__PURE__*/React.createElement(Icon, {
    name: props.icon,
    size: props.iconSize,
    color: props.iconColor ? props.iconColor : props.dark ? '#fff' : ''
  }) : null);
};

export default Avatar;
