// @assenti/rui-components/lib/inputMask/utils.js
export var phoneMaskFormatter = function phoneMaskFormatter(value) {
  if (value) {
    if (value.length <= 3) {
      return "(".concat(value, ")");
    } else if (value.length > 3 && value.length <= 6) {
      return "(".concat(value.slice(0, 3), ") ").concat(value.slice(3, value.length));
    } else if (value.length > 6 && value.length <= 8) {
      return "(".concat(value.slice(0, 3), ") ").concat(value.slice(3, 6), "-").concat(value.slice(6, value.length));
    } else {
      return "(".concat(value.slice(0, 3), ") ").concat(value.slice(3, 6), "-").concat(value.slice(6, 8), "-").concat(value.slice(8, 10));
    }
  } else return '';
};
export var iinMaskFormatter = function iinMaskFormatter(value) {
  if (value) {
    if (value.length <= 3) {
      return "".concat(value);
    } else if (value.length > 3 && value.length <= 6) {
      return "".concat(value.slice(0, 3), "-").concat(value.slice(3, value.length));
    } else if (value.length > 6 && value.length <= 9) {
      return "".concat(value.slice(0, 3), "-").concat(value.slice(3, 6), "-").concat(value.slice(6, value.length));
    } else {
      return "".concat(value.slice(0, 3), "-").concat(value.slice(3, 6), "-").concat(value.slice(6, 9), "-").concat(value.slice(9, 12));
    }
  } else return '';
};
export var cardMaskFormatter = function cardMaskFormatter(value) {
  if (value) {
    if (value.length <= 4) {
      return "".concat(value);
    } else if (value.length > 4 && value.length <= 8) {
      return "".concat(value.slice(0, 4), "-").concat(value.slice(4, value.length));
    } else if (value.length > 8 && value.length <= 12) {
      return "".concat(value.slice(0, 4), "-").concat(value.slice(4, 8), "-").concat(value.slice(8, value.length));
    } else {
      return "".concat(value.slice(0, 4), "-").concat(value.slice(4, 8), "-").concat(value.slice(8, 12), "-").concat(value.slice(12, 16));
    }
  } else return '';
};
export var cardIssueMaskFormatter = function cardIssueMaskFormatter(value) {
  if (value) {
    if (value.length <= 2) {
      return "".concat(value);
    } else {
      return "".concat(value.slice(0, 2), " / ").concat(value.slice(2, 4));
    }
  } else return '';
};
export var dateMaskFormatter = function dateMaskFormatter(value) {
  var delimiter = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : '.';

  if (value) {
    if (value.length <= 2) {
      return "".concat(value);
    } else if (value.length > 2 && value.length <= 4) {
      return "".concat(value.slice(0, 2)).concat(delimiter).concat(value.slice(2, value.length));
    } else {
      return "".concat(value.slice(0, 2)).concat(delimiter).concat(value.slice(2, 4)).concat(delimiter).concat(value.slice(4, 8));
    }
  } else return '';
};
export var timeMaskFormatter = function timeMaskFormatter(value) {
  var delimiter = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : ':';
  var withSeconds = arguments.length > 2 ? arguments[2] : undefined;

  if (value) {
    if (withSeconds) {
      if (value.length <= 2) {
        return "".concat(value);
      } else if (value.length > 2 && value.length <= 4) {
        return "".concat(value.slice(0, 2)).concat(delimiter).concat(value.slice(2, value.length));
      } else {
        return "".concat(value.slice(0, 2)).concat(delimiter).concat(value.slice(2, 4)).concat(delimiter).concat(value.slice(4, 6));
      }
    } else {
      if (value.length <= 2) {
        return "".concat(value);
      } else {
        return "".concat(value.slice(0, 2)).concat(delimiter).concat(value.slice(2, 4));
      }
    }
  } else return '';
};
