// @assenti/rui-components/lib/dock/index.js
import React, { useRef } from 'react';
import { CSSTransition } from 'react-transition-group';
import Avatar from '../avatar';
import Slider from '../slider';
import Divider from '../divider';
import { strinfigyClassObject } from '../utils';

var Dock = function Dock(props) {
  var dockItem = useRef();
  var className = {
    name: 'rui-dock',
    position: props.position ? props.position : 'bottom',
    tile: props.tile ? 'tile' : '',
    dark: props.dark ? 'dark' : ''
  };

  var handleClick = function handleClick(item) {
    if (props.onItemClick) {
      props.onItemClick(item);
    }
  };

  var getDockLength = function getDockLength() {
    if (props.length && (props.position === 'left' || props.position === 'right')) {
      return {
        height: props.length ? props.length : ''
      };
    } else if (props.length && (props.position === 'top' || props.position === 'bottom')) {
      return {
        width: props.length ? props.length : ''
      };
    }
  };

  var getVertical = function getVertical() {
    if (props.position === 'left' || props.position === 'right') {
      return true;
    } else {
      return false;
    }
  };

  return /*#__PURE__*/React.createElement("div", {
    className: "rui-dock-container ".concat(props.position ? props.position : '').trim()
  }, /*#__PURE__*/React.createElement(CSSTransition, {
    in: props.visible,
    timeout: 300,
    classNames: "dock-".concat(props.position),
    mountOnEnter: true,
    unmountOnExit: true
  }, /*#__PURE__*/React.createElement("div", {
    className: strinfigyClassObject(className),
    style: getDockLength()
  }, /*#__PURE__*/React.createElement(Slider, {
    items: props.items,
    dark: props.dark,
    length: props.length - 16,
    vertical: getVertical(),
    onItemClick: function onItemClick(item) {
      return handleClick(item);
    },
    render: function render(item) {
      return item.divider ? /*#__PURE__*/React.createElement(Divider, {
        vertical: true,
        dark: props.dark
      }) : /*#__PURE__*/React.createElement("div", {
        className: "rui-dock-icon",
        ref: dockItem
      }, /*#__PURE__*/React.createElement(Avatar, {
        img: item.img,
        icon: item.icon,
        iconColor: item.iconColor,
        width: props.itemSize ? props.itemSize : 60,
        height: props.itemSize ? props.itemSize : 60,
        transparent: true,
        iconSize: props.iconSize ? props.iconSize : 40
      }), item.title ? /*#__PURE__*/React.createElement("div", {
        className: "rui-dock-item-title"
      }, item.title) : null);
    }
  }))));
};

export default Dock;
