// react-rainbow-components/components/Calendar/pageObject/singleCalendar.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

/**
 * Calendar page object class.
 * @class
 * @tutorial calendar
 */
var PageSingleCalendar = /*#__PURE__*/function () {
  /**
   * Create a new PageCalendar page object.
   * @constructor
   * @param {string} rootElement - The selector of the PageCalendar root element.
   */
  function PageSingleCalendar(rootElement) {
    (0, _classCallCheck2["default"])(this, PageSingleCalendar);
    this.rootElement = rootElement;
  }
  /**
   * Clicks the previous month button element.
   * @method
   */


  (0, _createClass2["default"])(PageSingleCalendar, [{
    key: "clickPrevMonthButton",
    value: function clickPrevMonthButton() {
      $(this.rootElement).$$('button[data-id=button-icon-element]')[0].click();
    }
    /**
     * Clicks the next month button element.
     * @method
     */

  }, {
    key: "clickNextMonthButton",
    value: function clickNextMonthButton() {
      $(this.rootElement).$$('button[data-id=button-icon-element]')[1].click();
    }
    /**
     * Clicks the select year element.
     * @method
     */

  }, {
    key: "clickSelectYear",
    value: function clickSelectYear() {
      $(this.rootElement).$('select').click();
    }
    /**
     * Clicks the specific enabled day button element.
     * @method
     */

  }, {
    key: "clickDay",
    value: function clickDay(day) {
      var buttonEl = $(this.rootElement).$('table').$("button=".concat(day));
      if (buttonEl.isExisting()) buttonEl.click();
    }
    /**
     * Returns the text of the current selected month element.
     * @method
     * @returns {string}
     */

  }, {
    key: "getSelectedMonth",
    value: function getSelectedMonth() {
      return $(this.rootElement).$('h3[data-id=month]').getText();
    }
    /**
     * Returns the value of the select year element.
     * @method
     * @returns {string}
     */

  }, {
    key: "getSelectedYear",
    value: function getSelectedYear() {
      return $(this.rootElement).$('select').getValue();
    }
    /**
     * Returns the text of the current selected day element.
     * @method
     * @returns {string}
     */

  }, {
    key: "getSelectedDay",
    value: function getSelectedDay() {
      return $(this.rootElement).$('button[data-selected=true]').getText();
    }
    /**
     * Returns true when the specific day element is selected.
     * @method
     * @returns {string}
     */

  }, {
    key: "isDaySelected",
    value: function isDaySelected(day) {
      var buttonEl = $(this.rootElement).$('table').$("button=".concat(day));
      return buttonEl.isExisting() && buttonEl.getAttribute('data-selected') === 'true';
    }
    /**
     * Set the value of the year select element
     * @method
     * @param {string}
     */

  }, {
    key: "setYear",
    value: function setYear(value) {
      $(this.rootElement).$('select').selectByVisibleText(value);
    }
    /**
     * Returns true when the specific day button element has focus.
     * @method
     * @returns {bool}
     */

  }, {
    key: "isDayFocused",
    value: function isDayFocused(day) {
      var buttonEl = $(this.rootElement).$('table').$("button=".concat(day));
      return buttonEl.isExisting() && buttonEl.isFocused();
    }
    /**
     * Returns true when the previous month button element is disabled.
     * @method
     * @returns {bool}
     */

  }, {
    key: "isPrevMonthButtonDisabled",
    value: function isPrevMonthButtonDisabled() {
      var buttonEl = $(this.rootElement).$$('button[data-id=button-icon-element]')[0];
      return !buttonEl.isEnabled();
    }
    /**
     * Returns true when the next month button element is disabled.
     * @method
     * @returns {bool}
     */

  }, {
    key: "isNextMonthButtonDisabled",
    value: function isNextMonthButtonDisabled() {
      var buttonEl = $(this.rootElement).$$('button[data-id=button-icon-element]')[1];
      return !buttonEl.isEnabled();
    }
    /**
     * Returns true when the previous month button element has focus.
     * @method
     * @returns {bool}
     */

  }, {
    key: "isPrevMonthButtonFocused",
    value: function isPrevMonthButtonFocused() {
      var buttonEl = $(this.rootElement).$$('button[data-id=button-icon-element]')[0];
      return buttonEl.isExisting() && buttonEl.isFocused();
    }
    /**
     * Returns true when the next month button element has focus.
     * @method
     * @returns {bool}
     */

  }, {
    key: "isNextMonthButtonFocused",
    value: function isNextMonthButtonFocused() {
      var buttonEl = $(this.rootElement).$$('button[data-id=button-icon-element]')[1];
      return buttonEl.isExisting() && buttonEl.isFocused();
    }
    /**
     * Returns true when the year select element has focus.
     * @method
     * @returns {bool}
     */

  }, {
    key: "isYearSelectFocused",
    value: function isYearSelectFocused() {
      var selectEl = $(this.rootElement).$('select');
      return selectEl.isExisting() && selectEl.isFocused();
    }
  }]);
  return PageSingleCalendar;
}();

module.exports = PageSingleCalendar;
