// react-rainbow-components/components/ColorPicker/commons/hue/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = Hue;

var _react = _interopRequireWildcard(require("react"));

var _styled = _interopRequireDefault(require("./styled"));

var _color = require("../../../../styles/helpers/color");

var _context = require("../../context");

function Hue() {
  var _useContext = (0, _react.useContext)(_context.ColorPickerContext),
      hsv = _useContext.hsv,
      rgba = _useContext.rgba,
      tabIndex = _useContext.tabIndex,
      onChange = _useContext.onChange;

  var handleChange = function handleChange(event) {
    var value = parseInt(event.target.value, 10); // eslint-disable-next-line no-restricted-globals

    hsv[0] = isNaN(value) ? 0 : Math.max(0, Math.min(value, 360));
    var rgbColor = (0, _color.hsvToRgb)((0, _color.recomposeColor)({
      type: 'hsv',
      values: hsv
    }));
    var rgbaColor = (0, _color.rgbToRgba)(rgbColor, rgba[3]);
    onChange({
      hex: "#".concat((0, _color.rgbaToHex)(rgbaColor)),
      rgba: (0, _color.decomposeColor)(rgbaColor).values,
      hsv: hsv
    });
  };

  return /*#__PURE__*/_react["default"].createElement(_styled["default"], {
    value: hsv[0],
    min: 0,
    max: 360,
    onChange: handleChange,
    tabIndex: tabIndex
  });
}
