// @assenti/rui-components/lib/switch/index.js
import React from 'react';
import { Icon } from '../index';
import { strinfigyClassObject } from '../utils';

var Switch = function Switch(props) {
  var className = {
    name: 'rui-switch-container',
    size: props.size && props.size !== 'default' ? props.size : '',
    vertical: props.vertical ? 'vertical' : '',
    className: props.className ? props.className : ''
  };
  var classNameSwitch = {
    name: 'rui-switch',
    active: props.check ? 'active' : 'disactive',
    double: props.leftLabel || props.leftIcon ? 'double' : '',
    disabled: props.disabled ? 'disabled' : '',
    color: props.color ? props.color : ''
  };
  return /*#__PURE__*/React.createElement("div", {
    className: strinfigyClassObject(className)
  }, props.leftLabel && !props.leftIcon ? /*#__PURE__*/React.createElement("div", {
    className: "rui-switch-left-label"
  }, props.leftLabel) : '', props.leftIcon && !props.leftLabel ? /*#__PURE__*/React.createElement(Icon, {
    name: props.leftIcon,
    color: props.leftIconColor
  }) : '', /*#__PURE__*/React.createElement("div", {
    className: strinfigyClassObject(classNameSwitch),
    onClick: function onClick() {
      return !props.disabled ? props.onChange() : {};
    }
  }, /*#__PURE__*/React.createElement("span", null)), props.rightIcon && !props.rightLabel ? /*#__PURE__*/React.createElement(Icon, {
    name: props.rightIcon,
    color: props.rightIconColor
  }) : '', props.rightLabel && !props.rightIcon ? /*#__PURE__*/React.createElement("div", {
    className: "rui-switch-right-label"
  }, props.rightLabel) : '');
};

export default Switch;
